diff --git a/LunaSysMgr.upstart b/LunaSysMgr.upstart
index c197d80..e42b5a1 100644
--- a/LunaSysMgr.upstart
+++ b/LunaSysMgr.upstart
@@ -13,6 +13,8 @@ pre-start script
 	mkdir -p /var/luna/data
 	if grep -qs "qemu" /etc/hostname ; then
 		touch /var/luna/preferences/ran-first-use
+	elif grep -qs "tuna" /etc/hostname ; then
+		touch /var/luna/preferences/ran-first-use
 	fi
 	if [ ! -f /var/luna/preferences/ran-first-use ]
 	then
