diff --git a/Src/lunaui/cards/CardRoundedCornerShaderStage.h b/Src/lunaui/cards/CardRoundedCornerShaderStage.h
index 5bc2f8f..e31e1ea 100644
--- a/Src/lunaui/cards/CardRoundedCornerShaderStage.h
+++ b/Src/lunaui/cards/CardRoundedCornerShaderStage.h
@@ -24,11 +24,7 @@
 
 #include "Common.h"
 
-#if defined(HAVE_OPENGL) && defined(TARGET_DEVICE)
-#define USE_ROUNDEDCORNER_SHADER 1
-#else
 #undef USE_ROUNDEDCORNER_SHADER
-#endif
 
 #if defined(USE_ROUNDEDCORNER_SHADER)
 
diff --git a/Src/lunaui/cards/CardWindow.cpp b/Src/lunaui/cards/CardWindow.cpp
index c739321..e9022ac 100644
--- a/Src/lunaui/cards/CardWindow.cpp
+++ b/Src/lunaui/cards/CardWindow.cpp
@@ -26,11 +26,7 @@
 #include <PIpcBuffer.h>
 #include <PIpcChannel.h>
 
-#if defined(HAVE_OPENGL) && defined(TARGET_DEVICE)
-#define USE_ROUNDEDCORNER_SHADER 1
-#else
 #undef USE_ROUNDEDCORNER_SHADER
-#endif
 
 #if defined(USE_ROUNDEDCORNER_SHADER)
 #include <QtOpenGL/qglcustomshaderstage_p.h>
