diff --git a/Src/Main.cpp b/Src/Main.cpp
index 6f08397..1238e65 100644
--- a/Src/Main.cpp
+++ b/Src/Main.cpp
@@ -679,7 +679,7 @@ int main( int argc, char** argv)
 	// resolution may get picked up from the fb driver on arm
 	host->init(settings->displayWidth, settings->displayHeight);
 
-#if defined(TARGET_DEVICE)
+#if 0
 	pid_t animPid= spawnBootupAnimationProcess();
 	if(animPid < 0) { // failed to start the Animation process
 		return -1;
