diff --git a/LunaSysMgr.upstart b/LunaSysMgr.upstart
index c197d80..7443055 100644
--- a/LunaSysMgr.upstart
+++ b/LunaSysMgr.upstart
@@ -16,7 +16,7 @@ pre-start script
 	fi
 	if [ ! -f /var/luna/preferences/ran-first-use ]
 	then
-		echo " -u minimal -a com.palm.app.firstuse" > /var/luna/preferences/sysmgr-args
+		echo " -u minimal -a org.webosports.app.firstuse" > /var/luna/preferences/sysmgr-args
 	else
 		echo "" > /var/luna/preferences/sysmgr-args
 	fi
