diff --git a/conf/luna-a500.conf b/conf/luna-a500.conf
new file mode 100644
index 0000000..e78e76c
--- /dev/null
+++ b/conf/luna-a500.conf
@@ -0,0 +1,73 @@
+# @@@LICENSE
+#
+#      Copyright (c) 2010-2012 Hewlett-Packard Development Company, L.P.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# LICENSE@@@
+# @@@LICENSE
+#
+#      Copyright (c) 2010 Hewlett-Packard Development Company, L.P.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# LICENSE@@@
+[LaunchAtBoot]
+Applications=com.palm.app.phone;com.palm.app.email
+
+[General]
+ShowReticle=true
+DisplayWidth=1280
+DisplayHeight=800
+
+[UI]
+TabletUi=true
+DisplayUiRotates=true
+HomeButtonOrientationAngle=0
+ActiveCardWindowRatio=0.25
+NonActiveCardWindowRatio=0.20
+CardGroupRotFactor=90
+GapBetweenCardGroups=20
+UIScale=0.25
+TextScale=1.0
+LayoutScale=1.0
+IMEScale=1.0
+WebAppScale=1.0
+
+[Display]
+TurnOffAccelerometerWhenDimmed=false
+BrightnessOutdoorScale=255
+# We don't have support for ambient light sensor yet
+EnableALS=false
+
+[VirtualKeyboard]
+VirtualKeyboardEnabled=true
+
+[CoreNavi]
+# Device does not have a lightbar
+EnableLightBar=false
+
+[VirtualCoreNavi]
+VirtualCoreNaviEnabled=true
+VirtualCoreNaviHeight=64
+
diff --git a/device-known.pri b/device-known.pri
index 070866f..2c06578 100644
--- a/device-known.pri
+++ b/device-known.pri
@@ -70,3 +70,10 @@ contains(MACHINE_NAME, "tuna") {
     CONFIG_BUILD += opengl
     LIBS += -lqpalm
 }
+contains(MACHINE_NAME, "a500") {
+    DEFINES += MACHINE_A500
+    TARGET_TYPE = TARGET_DEVICE
+    CONFIG_BUILD += webosdevice nyx
+    CONFIG_BUILD += opengl
+    LIBS += -lqpalm
+}
