diff --git a/Src/lunaui/status-bar/SystemMenu.cpp b/Src/lunaui/status-bar/SystemMenu.cpp
index 87cc9d7..0726488 100644
--- a/Src/lunaui/status-bar/SystemMenu.cpp
+++ b/Src/lunaui/status-bar/SystemMenu.cpp
@@ -46,7 +46,7 @@
 #include "SystemUiController.h"
 
 #define SYS_UI_APP_ID      "com.palm.systemui"
-#define WIFI_PREFS_APP_ID  "com.palm.app.wifi"
+#define WIFI_PREFS_APP_ID  "org.webosports.app.settings"
 #define BLUETOOTH_PREFS_APP_ID  "com.palm.app.bluetooth"
 #define VPN_PREFS_APP_ID   "com.palm.app.vpn"
 
