From 39cd79259d951ee2507dae027cb17cf92ee7af3a Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 17 Dec 2018 09:02:57 +0000
Subject: [PATCH] tests: add vboxvideo to tested kernel modules

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Upstream-Status: Pending
---
 tests/util/kms.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/util/kms.c b/tests/util/kms.c
index 39a93866..a20eca56 100644
--- a/tests/util/kms.c
+++ b/tests/util/kms.c
@@ -124,6 +124,7 @@ const char *util_lookup_connector_type_name(unsigned int type)
 }
 
 static const char * const modules[] = {
+	"vboxvideo",
 	"i915",
 	"amdgpu",
 	"radeon",
