From cfc24b865d24766390f6754066e51f53e35e402a Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 24 Sep 2013 18:29:41 +0200
Subject: [PATCH 2/2] Set XDG_RUNTIME_DIR inside maliit-server statically

Upstream-Status: Inappropiate [other]

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 passthroughserver/main.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/passthroughserver/main.cpp b/passthroughserver/main.cpp
index 03d8169..d1c8e82 100644
--- a/passthroughserver/main.cpp
+++ b/passthroughserver/main.cpp
@@ -116,6 +116,8 @@ QSharedPointer<Maliit::AbstractPlatform> createPlatform()
 
 int main(int argc, char **argv)
 {
+    setenv("XDG_RUNTIME_DIR", "/tmp/luna-session", 0);
+
     qInstallMessageHandler(outputMessagesToStdErr);
 
     // QT_IM_MODULE, MApplication and QtMaemo5Style all try to load
-- 
1.8.1.2

