From f365e8241256c408b1d22949024c814d80710883 Mon Sep 17 00:00:00 2001
From: Herrie <github.com@herrie.org>
Date: Sun, 28 Feb 2016 11:39:16 +0100
Subject: [PATCH] Use Google as default and add API key

Upstream-Status: Pending
---
 data/geoclue.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/geoclue.conf.in b/data/geoclue.conf.in
index 8006cda..9d7a683 100644
--- a/data/geoclue.conf.in
+++ b/data/geoclue.conf.in
@@ -58,7 +58,7 @@ enable=true
 #
 # https://developers.google.com/maps/documentation/geolocation/policies
 #
-#url=https://www.googleapis.com/geolocation/v1/geolocate?key=YOUR_KEY
+url=https://www.googleapis.com/geolocation/v1/geolocate?key=AIzaSyBSTvlMEfLkRzgdZbqhv7PsA2MulWjgYNs
 
 # Submit data to Mozilla Location Service
 # If set to true, geoclue will automatically submit network data to Mozilla
