From 07ab5b55f878fe67b16b7bbdccf9166a5af415d5 Mon Sep 17 00:00:00 2001
From: Herrie <github.com@herrie.org>
Date: Sun, 28 Feb 2016 11:39:16 +0100
Subject: [PATCH] Use Google as default and add API key

Upstream-Status: Pending
---
 data/geoclue.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/geoclue.conf.in b/data/geoclue.conf.in
index 0eb59cd..ee9629a 100644
--- a/data/geoclue.conf.in
+++ b/data/geoclue.conf.in
@@ -60,7 +60,7 @@ enable=@default_wifi_enable@
 #
 # https://developers.google.com/maps/documentation/geolocation/policies
 #
-#url=https://www.googleapis.com/geolocation/v1/geolocate?key=YOUR_KEY
+url=https://www.googleapis.com/geolocation/v1/geolocate?key=AIzaSyBSTvlMEfLkRzgdZbqhv7PsA2MulWjgYNs
 
 # Submit data to WiFi geolocation service
 # If set to true, geoclue will automatically submit network data
