From 68cfdbff39ab803d375c22bc9d9b4c582c8115a6 Mon Sep 17 00:00:00 2001
From: Herrie <github.com@herrie.org>
Date: Sun, 28 Feb 2016 11:39:16 +0100
Subject: [PATCH] Use Google as default and add API key

---
 data/geoclue.conf.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/geoclue.conf.in b/data/geoclue.conf.in
index 5009ebf..93665d4 100644
--- a/data/geoclue.conf.in
+++ b/data/geoclue.conf.in
@@ -18,13 +18,13 @@ enable=true
 
 # URL to the wifi geolocation service. The key can currenty be anything, just
 # needs to be present but that is likely going to change in future.
-url=https://location.services.mozilla.com/v1/geolocate?key=geoclue
+#url=https://location.services.mozilla.com/v1/geolocate?key=geoclue
 
 # To use the Google geolocation service instead of mozilla's, simply uncomment
 # this url while changing API_KEY to your Google API key and comment out or
 # remove the url above.
 #
-#url=https://www.googleapis.com/geolocation/v1/geolocate?key=YOUR_KEY
+url=https://www.googleapis.com/geolocation/v1/geolocate?key=AIzaSyBSTvlMEfLkRzgdZbqhv7PsA2MulWjgYNs
 
 # Submit data to Mozilla Location Service
 # If set to true, geoclue will automatically submit network data to Mozilla
-- 
2.5.3.windows.1

