Upstream-Status: Pending

--- bluez-5.43/tools/hciattach_bcm43xx.c	2015-03-11 10:01:57.000000000 +0000
+++ ../bluez-5.43/tools/hciattach_bcm43xx.c	2017-11-04 12:24:29.631280622 +0000
@@ -329,14 +329,14 @@
 			ret = bcm43xx_locate_patch(path, chip_name, location);
 			if (!ret)
 				break;
-		} else if (!strncmp(chip_name, entry->d_name, strlen(chip_name))) {
+		} else if (!strncasecmp(chip_name, entry->d_name, strlen(chip_name))) {
 			unsigned int name_len = strlen(entry->d_name);
 			size_t curs_ext = name_len - sizeof(FW_EXT) + 1;
 
 			if (curs_ext > name_len)
 				break;
 
-			if (strncmp(FW_EXT, &entry->d_name[curs_ext], sizeof(FW_EXT)))
+			if (strncasecmp(FW_EXT, &entry->d_name[curs_ext], sizeof(FW_EXT)))
 				break;
 
 			/* found */
