diff --git a/recipes-qt/maliit/maliit-framework-qt5/maliit-server.service b/recipes-qt/maliit/maliit-framework-qt5/maliit-server.service
index 3bf3de1..04ce960 100644
--- a/recipes-qt/maliit/maliit-framework-qt5/maliit-server.service
+++ b/recipes-qt/maliit/maliit-framework-qt5/maliit-server.service
@@ -5,6 +5,7 @@ Requires=luna-next.service
 
 [Service]
 Type=dbus
+EnvironmentFile=-/etc/luna-next/maliit-env.conf
 BusName=org.maliit.server
 ExecStart=/usr/bin/maliit-server
 
diff --git a/recipes-qt/maliit/maliit-framework-qt5_git.bbappend b/recipes-qt/maliit/maliit-framework-qt5_git.bbappend
index 8ac0d44..0036acd 100644
--- a/recipes-qt/maliit/maliit-framework-qt5_git.bbappend
+++ b/recipes-qt/maliit/maliit-framework-qt5_git.bbappend
@@ -5,6 +5,7 @@ SRC_URI += " \
     file://0003-Add-webos-platform-implementation-to-set-correct-win.patch \
     file://maliit-server.conf \
     file://maliit-server.service \
+    file://maliit-env.conf \
 "
 
 inherit systemd
@@ -16,6 +17,9 @@ do_install_append() {
     install -d ${D}${sysconfdir}/dbus-1/system.d
     install -m 0644 ${WORKDIR}/maliit-server.conf ${D}${sysconfdir}/dbus-1/system.d/
 
+    install -d ${D}${sysconfdir}/luna-next
+    install -m 0644 ${WORKDIR}/maliit-env.conf ${D}${sysconfdir}/luna-next/
+
     install -d ${D}${systemd_unitdir}/system
     install -m 0644 ${WORKDIR}/maliit-server.service ${D}${systemd_unitdir}/system/
 }
