From 1b3cf52ad93f35a20ae9ce1e4934d9c7b0a6b975 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 3 Apr 2019 11:07:33 +0000
Subject: [PATCH] logind.conf.in: disable key handling in logind

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/login/logind.conf.in | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/login/logind.conf.in b/src/login/logind.conf.in
index 1029e29bc7..3130c7a92e 100644
--- a/src/login/logind.conf.in
+++ b/src/login/logind.conf.in
@@ -18,12 +18,12 @@
 #KillOnlyUsers=
 #KillExcludeUsers=root
 #InhibitDelayMaxSec=5
-#HandlePowerKey=poweroff
-#HandleSuspendKey=suspend
-#HandleHibernateKey=hibernate
-#HandleLidSwitch=suspend
-#HandleLidSwitchExternalPower=suspend
-#HandleLidSwitchDocked=ignore
+HandlePowerKey=ignore
+HandleSuspendKey=ignore
+HandleHibernateKey=ignore
+HandleLidSwitch=ignore
+HandleLidSwitchExternalPower=ignore
+HandleLidSwitchDocked=ignore
 #PowerKeyIgnoreInhibited=no
 #SuspendKeyIgnoreInhibited=no
 #HibernateKeyIgnoreInhibited=no
