From ceb72f191dee3363884e9183e4d30e5064209f33 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 21 Oct 2023 11:39:18 +0000
Subject: [PATCH] core.pro: use QT6_INSTALL_LIBDIR when provided

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Upstream-Status: Submitted [https://github.com/sailfishos/sensorfw/pull/20]
---
 core/core.pro | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/core.pro b/core/core.pro
index 5239e36..c593e75 100644
--- a/core/core.pro
+++ b/core/core.pro
@@ -15,7 +15,11 @@ SENSORFW_INCLUDEPATHS = .. \
 DEPENDPATH += $$SENSORFW_INCLUDEPATHS
 INCLUDEPATH += $$SENSORFW_INCLUDEPATHS
 
-DEFINES += PLUGIN_DIRECTORY=\"\\\"$$[QT_INSTALL_LIBS]/sensord-qt$${QT_MAJOR_VERSION}\\\"\"
+!isEmpty(QT6_INSTALL_LIBDIR) {
+    DEFINES += PLUGIN_DIRECTORY=\"\\\"$${QT6_INSTALL_LIBDIR}/sensord-qt$${QT_MAJOR_VERSION}\\\"\"
+} else {
+    DEFINES += PLUGIN_DIRECTORY=\"\\\"$$[QT_INSTALL_LIBS]/sensord-qt$${QT_MAJOR_VERSION}\\\"\"
+}
 
 QMAKE_LIBDIR_FLAGS += -L../datatypes\
                       -lsensordatatypes-qt$${QT_MAJOR_VERSION}
-- 
2.34.1

