From a01723b06da2441806baf3f323eb49b801c9b2c6 Mon Sep 17 00:00:00 2001
From: Nikolay Nizov <nizovn@gmail.com>
Date: Mon, 23 Feb 2015 00:07:36 +0300
Subject: [PATCH] sensorfwd.service: Change wanted target to multi-user

Signed-off-by: Nikolay Nizov <nizovn@gmail.com>
---
 rpm/sensorfwd.service | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/rpm/sensorfwd.service b/rpm/sensorfwd.service
index ecbf1d5..4c8fb92 100644
--- a/rpm/sensorfwd.service
+++ b/rpm/sensorfwd.service
@@ -14,5 +14,4 @@ Restart=always
 RestartSec=1
 
 [Install]
-WantedBy=graphical.target
-
+WantedBy=multi-user.target
-- 
2.0.5

