From 5aa80fca57482e03003e72279d460723c4786baf Mon Sep 17 00:00:00 2001
From: Nikolay Nizov <nizovn@gmail.com>
Date: Sat, 11 Apr 2015 12:31:18 +0300
Subject: [PATCH] Set okToStop to false to prevent deadlock when stopping
 sensors

Signed-off-by: Nikolay Nizov <nizovn@gmail.com>
---
 core/hybrisadaptor.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/hybrisadaptor.cpp b/core/hybrisadaptor.cpp
index 8cd6409..e800369 100644
--- a/core/hybrisadaptor.cpp
+++ b/core/hybrisadaptor.cpp
@@ -202,7 +202,8 @@ void HybrisManager::stopReader(HybrisAdaptor *adaptor)
 {
     QList <HybrisAdaptor *> list;
     list = registeredAdaptors.values();
-    bool okToStop = true;
+    // never stop reading thread because otherwise it will be blocked in poll, and never return
+    bool okToStop = false;
 
     for (int i = 0; i < list.count(); i++) {
         if (list.at(i) == adaptor && !list.at(i)->isRunning()) {
-- 
2.0.5

