From 352e05cdba00b8f225f7185d9bb9f464f34c59da Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sun, 14 Oct 2012 16:40:08 +0200
Subject: [PATCH 2/2] Use wakelock daemon to handle device state while in
 suspend

---
 scripts/public/suspend_action | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/public/suspend_action b/scripts/public/suspend_action
index 27d2e11..29b6d96 100755
--- a/scripts/public/suspend_action
+++ b/scripts/public/suspend_action
@@ -24,6 +24,9 @@ PmLogCtl -s logkv sleepdscript info "sleepd: System going to sleep now"
 
 # Initiate suspend
 echo mem > /sys/power/state
+/usr/sbin/wakelockd
+echo on > /sys/power/state
+
 
 PmLogCtl -s logkv sleepdscript info "sleepd: System woke up"
 
-- 
1.8.5.3

