From 3047146b0a4bb776e44d37f8915bac1a9f0b7259 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sun, 14 Oct 2012 16:40:08 +0200
Subject: [PATCH 2/2] Use wakelock daemon to handle device state while in
 suspend

---
 scripts/public/suspend_action | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/scripts/public/suspend_action b/scripts/public/suspend_action
index 27d2e11..49125f8 100755
--- a/scripts/public/suspend_action
+++ b/scripts/public/suspend_action
@@ -22,8 +22,10 @@ PmLogCtl -s def sleepdscript info
 
 PmLogCtl -s logkv sleepdscript info "sleepd: System going to sleep now"
 
-# Initiate suspend
-echo mem > /sys/power/state
+# Wakelockd will do all the suspend handling for us and exit once we should wake up again.
+# Suspend is asynchronous as we're normally placed on an android based system which uses
+# wakelocks.
+/usr/sbin/wakelockd
 
 PmLogCtl -s logkv sleepdscript info "sleepd: System woke up"
 
-- 
1.9.1

