From ce90bc717010a13bddd4d2ef15a8b406dea43ec0 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sun, 14 Oct 2012 16:40:08 +0200
Subject: [PATCH] Use wakelock daemon to handle device state while in suspend

---
 scripts/public/suspend_action |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/public/suspend_action b/scripts/public/suspend_action
index 0557b74..de5e5d5 100755
--- a/scripts/public/suspend_action
+++ b/scripts/public/suspend_action
@@ -22,6 +22,8 @@ echo -e "sleepd: System going to sleep now\n" | wall
 
 # Initiate suspend
 echo mem > /sys/power/state
+/usr/sbin/wakelockd
+echo on > /sys/power/state
 
 echo -e "sleepd: System woke up.\n" | wall 
 
-- 
1.7.9.5

