From d0db154e3651a2ebc1056a9850e56926bc3b0550 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 19 Dec 2014 18:47:09 +0100
Subject: [PATCH] replace cjson with json-c

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 CMakeLists.txt | 8 ++++----
 README.md      | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e570aa7..edee032 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,9 +40,9 @@ pkg_check_modules(LUNASERVICE REQUIRED luna-service2)
 include_directories(${LUNASERVICE_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${LUNASERVICE_CFLAGS})
 
-pkg_check_modules(CJSON cjson)
-include_directories(${CJSON_INCLUDE_DIRS})
-webos_add_compiler_flags(ALL ${CJSON_CFLAGS})
+pkg_check_modules(JSON json-c)
+include_directories(${JSON_INCLUDE_DIRS})
+webos_add_compiler_flags(ALL ${JSON_CFLAGS})
 
 pkg_check_modules(DB8 REQUIRED db8>=3)
 include_directories(${DB8_INCLUDE_DIRS})
@@ -65,7 +65,7 @@ target_link_libraries(configurator
                     ${DB8_LDFLAGS}
                    	${GLIB2_LDFLAGS}
                    	${LUNASERVICE_LDFLAGS}
-                   	${CJSON_LDFLAGS}
+                   	${JSON_LDFLAGS}
                         ${PTHREAD}
                     ${PMLOG_LDFLAGS}
                    )
diff --git a/README.md b/README.md
index 49aa08d..1b160e0 100644
--- a/README.md
+++ b/README.md
@@ -13,7 +13,7 @@ Dependencies
 ============
 Below are the tools and libraries required to build.
 
-	- openwebos/cjson 1.8.0
+	- json-c 0.11
 	- cmake 2.8.7 
 	- glib-2.0
 	- openwebos/Db8 libraries (libmojocore, libmojoluna)
-- 
2.2.1

