From dcd2676a3b97005b759e12941921de6a8da5f301 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 21 Jan 2023 09:10:47 +0000
Subject: [PATCH] gst_v4l2_fill_lists: abort if type is invalid

---
 sys/v4l2/v4l2_calls.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sys/v4l2/v4l2_calls.c b/sys/v4l2/v4l2_calls.c
index c13b142..8184c7c 100644
--- a/sys/v4l2/v4l2_calls.c
+++ b/sys/v4l2/v4l2_calls.c
@@ -169,6 +169,9 @@ gst_v4l2_fill_lists (GstV4l2Object * v4l2object)
         (guint64) input.std);
     GST_LOG_OBJECT (e, "   status:    %08x", input.status);
 
+    // invalid input type: abort
+    if (input.type == 0) return FALSE;
+
     v4l2channel = g_object_new (GST_TYPE_V4L2_TUNER_CHANNEL, NULL);
     channel = GST_TUNER_CHANNEL (v4l2channel);
     channel->label = g_strdup ((const gchar *) input.name);
