From 21be16fe9411c23143c88548da7cc89cf80e73a6 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Thu, 20 Nov 2014 19:21:13 +0100
Subject: [PATCH 1/4] Enable the various modem plugins we support again

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 ofono/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ofono/Makefile.am b/ofono/Makefile.am
index 156e420..4f3de65 100644
--- a/ofono/Makefile.am
+++ b/ofono/Makefile.am
@@ -396,7 +396,6 @@ builtin_sources += drivers/cdmamodem/cdmamodem.h \
 			drivers/cdmamodem/connman.c
 endif
 
-if !RILMODEM
 builtin_modules += g1
 builtin_sources += plugins/g1.c
 
-- 
2.1.4

