From 9b1812dbaa553542a9be203137039d89eae977d7 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 13 Jul 2019 13:30:38 +0000
Subject: [PATCH] Enable the various modem plugins we support again

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 ofono/Makefile.am | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/ofono/Makefile.am b/ofono/Makefile.am
index bed21ae1..171f18f3 100644
--- a/ofono/Makefile.am
+++ b/ofono/Makefile.am
@@ -493,7 +493,6 @@ builtin_sources += drivers/cdmamodem/cdmamodem.h \
 			drivers/cdmamodem/connman.c
 endif
 
-if !RILMODEM
 builtin_modules += g1
 builtin_sources += plugins/g1.c
 
@@ -580,10 +579,6 @@ builtin_sources += plugins/ublox.c
 
 builtin_modules += xmm7xxx
 builtin_sources += plugins/xmm7xxx.c
-endif
-
-builtin_modules += connman
-builtin_sources += plugins/connman.c
 
 builtin_modules += mnclength
 builtin_sources += plugins/mnclength.c
-- 
2.17.1
