From 9b1812dbaa553542a9be203137039d89eae977d7 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 13 Jul 2019 13:30:38 +0000
Subject: [PATCH] Enable the various modem plugins we support again

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 ofono/Makefile.am | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/ofono/Makefile.am b/ofono/Makefile.am
index f59f7a7e..b1a71818 100644
--- a/ofono/Makefile.am
+++ b/ofono/Makefile.am
@@ -493,7 +493,6 @@ builtin_sources += drivers/cdmamodem/cdmamodem.h \
 			drivers/cdmamodem/connman.c
 endif
 
-if !RILMODEM
 builtin_modules += g1
 builtin_sources += plugins/g1.c
 
@@ -569,9 +568,6 @@ builtin_sources += plugins/samsung.c
 builtin_modules += sim900
 builtin_sources += plugins/sim900.c
 
-builtin_modules += connman
-builtin_sources += plugins/connman.c
-
 builtin_modules += telit
 builtin_sources += plugins/telit.c
 
@@ -583,7 +579,6 @@ builtin_sources += plugins/ublox.c
 
 builtin_modules += xmm7xxx
 builtin_sources += plugins/xmm7xxx.c
-endif
 
 builtin_modules += connman
 builtin_sources += plugins/connman.c
-- 
2.17.0

