From 6d63ad35c8ea627057797ba91fddd3d952cbc58c Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sat, 12 Oct 2013 16:57:22 +0200
Subject: [PATCH 3/3] Disable backtrace cause linking to libdl fails

Upstream-Status: Inappropiate [configuration]

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 ofono/src/log.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ofono/src/log.c b/ofono/src/log.c
index a8cdaa8..2f398d8 100644
--- a/ofono/src/log.c
+++ b/ofono/src/log.c
@@ -113,6 +113,7 @@ void ofono_debug(const char *format, ...)
 	va_end(ap);
 }
 
+#if 0
 static void print_backtrace(unsigned int offset)
 {
 	void *frames[99];
@@ -214,12 +215,15 @@ static void print_backtrace(unsigned int offset)
 	close(outfd[1]);
 	close(infd[0]);
 }
+#endif
 
 static void signal_handler(int signo)
 {
 	ofono_error("Aborting (signal %d) [%s]", signo, program_exec);
 
+#if 0
 	print_backtrace(2);
+#endif
 
 	exit(EXIT_FAILURE);
 }
-- 
2.1.4

