From 157f89adb1d781c9746e376215c371060788c3e1 Mon Sep 17 00:00:00 2001
From: Joey Hewitt <joey@joeyhewitt.com>
Date: Fri, 19 Jul 2019 22:02:07 +0200
Subject: [PATCH] support smdpkt

Upstream-Status: Pending
---
 plugins/udevng.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/plugins/udevng.c b/plugins/udevng.c
index 8bdcf0d..ec5d49f 100644
--- a/plugins/udevng.c
+++ b/plugins/udevng.c
@@ -252,6 +252,7 @@ static gboolean setup_gobi(struct modem_info *modem)
 
 	DBG("%s", modem->syspath);
 
+	if (modem->type != MODEM_TYPE_SERIAL) {
 	for (list = modem->devices; list; list = list->next) {
 		const struct device_info *info = list->data;
 		const char *subsystem =
@@ -284,6 +285,9 @@ static gboolean setup_gobi(struct modem_info *modem)
 			}
 		}
 	}
+	} else {
+		qmi = modem->serial;
+	}
 
 	if (qmi == NULL || mdm == NULL || net == NULL)
 		return FALSE;
@@ -2205,6 +2209,8 @@ static void enumerate_devices(struct udev *context)
 		return;
 
 	udev_enumerate_add_match_subsystem(enumerate, "tty");
+	udev_enumerate_add_match_subsystem(enumerate, "smdpkt");
+	udev_enumerate_add_match_subsystem(enumerate, "rpmsg");
 	udev_enumerate_add_match_subsystem(enumerate, "usb");
 	udev_enumerate_add_match_subsystem(enumerate, "usbmisc");
 	udev_enumerate_add_match_subsystem(enumerate, "net");
