diff --git a/lib/sqlite3.js b/lib/sqlite3.js
index be9205c..4808ddf 100644
--- a/lib/sqlite3.js
+++ b/lib/sqlite3.js
@@ -1,8 +1,5 @@
 var path = require('path');
-var pkg = require('../package.json');
-var module_path = path.join(
-     path.join(__dirname,'../' + pkg.binary.module_path),
-     pkg.binary.module_name + '.node');
+var module_path = '/usr/lib/nodejs/node_sqlite3.node';
 var binding = require(module_path);
 var sqlite3 = module.exports = exports = binding;
 var util = require('util');
