From 3efb6b376fd0ca2266076ebaa71d679d370b06f9 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 3 Feb 2017 22:54:54 +0100
Subject: [PATCH 2/3] Fix 4.4.2 hwcomposer build

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 hwcomposer/hwcomposer_backend_v0.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hwcomposer/hwcomposer_backend_v0.cpp b/hwcomposer/hwcomposer_backend_v0.cpp
index d40a6b3..6eed37c 100644
--- a/hwcomposer/hwcomposer_backend_v0.cpp
+++ b/hwcomposer/hwcomposer_backend_v0.cpp
@@ -40,8 +40,9 @@
 ****************************************************************************/
 #include <android-version.h>
 #if ANDROID_VERSION_MAJOR <= 4
-#include "hwcomposer_backend_v0.h"
+#include <hardware/hwcomposer_defs.h>
 #ifdef HWC_DEVICE_API_VERSION_0_1
+#include "hwcomposer_backend_v0.h"
 
 
 HwComposerBackend_v0::HwComposerBackend_v0(hw_module_t *hwc_module, hw_device_t *hw_device)
-- 
2.10.2

