From 50efefd89bf7a3bf220065f133796a3d018a0fd7 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Wed, 22 May 2024 21:18:11 +0000
Subject: [PATCH] Remove useless machine dependency

Upstream-Status: Pending

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e28c220..9e3e3b4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,6 @@ include(webOS/LegacyDefines)
 
 webos_modules_init(1 0 0 QUALIFIER RC2)
 webos_component(2 0 1)
-webos_machine_impl_dep()
 webos_add_compiler_flags(ALL -std=c++11 -Wall -Wconversion -DMOJ_LINUX)
 
 include(FindPkgConfig)
