From 0100934140c848547f76d8904d96758606208739 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Fri, 7 Jun 2019 08:57:39 +0000
Subject: [PATCH] Disable gtk-doc

Upstream-Status: Pending
---
 Makefile.am  | 2 +-
 configure.ac | 2 --
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index f87db22..bbb0feb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = data src docs
+SUBDIRS = data src
 
 DISTCHECK_CONFIGURE_FLAGS =								\
 	--enable-gtk-doc								\
diff --git a/configure.ac b/configure.ac
index 6786d18..9ff36f6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -64,7 +64,5 @@ AC_CONFIG_FILES([
 	Makefile
 	data/Makefile
 	src/Makefile
-	docs/Makefile
-	docs/version.xml
 ])
 AC_OUTPUT
-- 
2.17.0

