diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 408d31f..9063a5f 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -24,7 +24,7 @@ include(webOS/webOS)
 webos_modules_init(1 0 0 QUALIFIER RC2)
 webos_component(2 0 0)
 
-webos_add_compiler_flags(ALL "-DMOJ_LINUX -DMOJ_X86")
+webos_add_compiler_flags(ALL "-DMOJ_LINUX -DMOJ_ARM")
 webos_add_compiler_flags(ALL "-O2 -ggdb -Wuninitialized -Winit-self -Wno-error=strict-aliasing")
 
 webos_machine_dep()
