diff --git a/smtp/CMakeLists.txt b/smtp/CMakeLists.txt
index 1b6536d..7e47751 100644
--- a/smtp/CMakeLists.txt
+++ b/smtp/CMakeLists.txt
@@ -24,7 +24,7 @@ include(webOS/webOS)
 webos_modules_init(1 0 0 QUALIFIER RC2)
 webos_component(2 0 0)
 
-webos_add_compiler_flags(ALL "-DMOJ_LINUX -DMOJ_X86")
+webos_add_compiler_flags(ALL "-DMOJ_LINUX -DMOJ_ARM")
 webos_add_compiler_flags(ALL "-O -ggdb -Wuninitialized -Winit-self")
 
 webos_add_linker_options(ALL "--unresolved-symbols=ignore-in-shared-libs -rpath-link ${WEBOS_INSTALL_ROOT}/lib")
