From 9d992dcf9171c74caea1722ae88c158319a669c9 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Wed, 22 May 2024 21:01:52 +0000
Subject: [PATCH] Remove webos_machine_impl_dep

Upstream-Status: Pending

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6c9ceb5..590a32c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,6 @@ project(sandbox CXX)
 include(webOS/webOS)
 webos_modules_init(1 0 0 QUALIFIER RC2)
 webos_component(2 0 0)
-webos_machine_impl_dep()
 
 webos_build_pkgconfig()
 include_directories(include/public)
