From ca9fa1e6a5340836484784d498a38d1b8103bd98 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 16 Feb 2017 19:11:33 +0000
Subject: [PATCH 1/2] hooks: include mm hooks only when WANT_LINKER_MM is
 defined

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 hybris/common/hooks.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/hybris/common/hooks.c b/hybris/common/hooks.c
index 9b7df48..4c119ec 100644
--- a/hybris/common/hooks.c
+++ b/hybris/common/hooks.c
@@ -2874,8 +2874,10 @@ static void* __hybris_get_hooked_symbol(const char *sym, const char *requester)
 
     /* Allow newer hooks to override those which are available for all versions */
     key.name = sym;
+#if defined(WANT_LINKER_MM)
     if (get_android_sdk_version() > 21)
         found = bsearch(&key, hooks_mm, HOOKS_SIZE(hooks_mm), sizeof(hooks_mm[0]), hook_cmp);
+#endif
     if (!found)
         found = bsearch(&key, hooks_common, HOOKS_SIZE(hooks_common), sizeof(hooks_common[0]), hook_cmp);
 
-- 
2.7.4

