From e6627195d89cf117e3f6475db1dddb5bfe7a256c Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 8 Apr 2023 13:46:44 +0000
Subject: [PATCH] QWaylandDisplay: don't ignore 'wayland' QT_IM_MODULE

Upstream-Status: Pending
Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/client/qwaylanddisplay_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/client/qwaylanddisplay_p.h b/src/client/qwaylanddisplay_p.h
index 2fd176d8..7610ca88 100644
--- a/src/client/qwaylanddisplay_p.h
+++ b/src/client/qwaylanddisplay_p.h
@@ -360,7 +360,7 @@ private:
 
     bool mWaylandInputContextRequested = [] () {
         const auto requested = QPlatformInputContextFactory::requested();
-        return requested.isEmpty() || requested.contains(QLatin1String(WAYLAND_IM_KEY));
+        return requested.isEmpty() /* || requested.contains(QLatin1String(WAYLAND_IM_KEY)) */;
     }();
     QStringList mTextInputManagerList;
     int mTextInputManagerIndex = INT_MAX;
