From f67fe919f2e35f39826858ff0d190f95213e4e90 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 8 Apr 2023 13:46:44 +0000
Subject: [PATCH] QWaylandDisplay: don't ignore 'wayland' QT_IM_MODULE

Upstream-Status: Pending
Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/client/qwaylanddisplay_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/client/qwaylanddisplay_p.h b/src/client/qwaylanddisplay_p.h
index 473016f1..4234033a 100644
--- a/src/client/qwaylanddisplay_p.h
+++ b/src/client/qwaylanddisplay_p.h
@@ -278,7 +278,7 @@ private:
 
     bool mClientSideInputContextRequested = [] () {
         const QString& requested = QPlatformInputContextFactory::requested();
-        return !requested.isEmpty() && requested != QLatin1String("wayland");
+        return !requested.isEmpty() /* && requested != QLatin1String("wayland") */;
     }();
     QStringList mTextInputManagerList;
     int mTextInputManagerIndex = INT_MAX;
-- 
2.34.1

