diff --git a/Src/webbase/WebAppManager.cpp b/Src/webbase/WebAppManager.cpp
index bab2615..0641bec 100644
--- a/Src/webbase/WebAppManager.cpp
+++ b/Src/webbase/WebAppManager.cpp
@@ -217,7 +217,7 @@ WebAppManager::WebAppManager()
     setenv("QT_PLUGIN_PATH", "/usr/plugins", 1);
     setenv("QT_DEBUG_PLUGINS", "1", 1);
 
-#if defined(TARGET_DEVICE)
+#if defined(TARGET_DEVICE) && !defined(MACHINE_TUNA) && !defined(MACHINE_GROUPER) && !defined(MACHINE_A500)
     static const char *argv[] = { "./WebAppManager", "-platform", "webos", NULL };
 #else
     static const char *argv[] = { "./WebAppManager", "-platform", "minimal", NULL };
diff --git a/device-known.pri b/device-known.pri
index 070866f..ce8c18d 100644
--- a/device-known.pri
+++ b/device-known.pri
@@ -70,3 +70,11 @@ contains(MACHINE_NAME, "tuna") {
     CONFIG_BUILD += opengl
     LIBS += -lqpalm
 }
+contains(MACHINE_NAME, "a500") {
+    DEFINES += MACHINE_A500 HAS_DISPLAY_TIMEOUT
+    TARGET_TYPE = TARGET_DEVICE
+    CONFIG_BUILD += webosdevice nyx
+    CONFIG_BUILD += opengl
+    LIBS += -lqpalm
+}
+
