diff --git a/Src/webbase/WebAppManager.cpp b/Src/webbase/WebAppManager.cpp
index bab2615..0641bec 100644
--- a/Src/webbase/WebAppManager.cpp
+++ b/Src/webbase/WebAppManager.cpp
@@ -217,7 +217,7 @@ WebAppManager::WebAppManager()
     setenv("QT_PLUGIN_PATH", "/usr/plugins", 1);
     setenv("QT_DEBUG_PLUGINS", "1", 1);
 
-#if defined(TARGET_DEVICE)
+#if defined(TARGET_DEVICE) && !defined(MACHINE_TUNA)
     static const char *argv[] = { "./WebAppManager", "-platform", "webos", NULL };
 #else
     static const char *argv[] = { "./WebAppManager", "-platform", "minimal", NULL };
