From e4b8836f5f670996a00c186e4c109b0a7eb524e6 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 16 Sep 2015 17:04:01 +0200
Subject: [PATCH] psl_build_config.h: don't use extern with inline

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/psl_build_config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/psl_build_config.h b/src/psl_build_config.h
index 0636d6c..a09c5f3 100644
--- a/src/psl_build_config.h
+++ b/src/psl_build_config.h
@@ -40,7 +40,7 @@ extern "C" {
  */
 #ifndef PSL_CONFIG_INLINE_FUNC
     #ifdef __GNUC__
-        #define PSL_CONFIG_INLINE_FUNC  extern __inline
+        #define PSL_CONFIG_INLINE_FUNC  inline
     #else
         #error "I don't know how to declare inline funcs in this build environment"
     #endif
-- 
2.5.1

