diff --git a/CMakeLists.txt b/CMakeLists.txt
index 41e460a..9ac93a6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,12 +58,12 @@ if(${MJDB} STREQUAL MJDB-NOTFOUND)
 	message(FATAL_ERROR "Failed to find libmojodb. Please install openwebos/db8")
 endif()
 
-find_path(MJ_INC NAMES core/MojErr.h HINTS /include/mojodb )
+find_path(MJ_INC NAMES mojodb/core/MojErr.h)
 if(MJ_INC STREQUAL "MJ_INC-NOTFOUND")
 	message(FATAL_ERROR "Failed to find db8 include files")
 endif()
-include_directories(${MJ_INC})
-webos_add_compiler_flags(ALL -I${MJ_INC})
+include_directories(${MJ_INC}/mojodb)
+webos_add_compiler_flags(ALL -I${MJ_INC}/mojodb)
 
 # There is a call to system("initctl ...") in ServiceApp.cpp which is compiled in for
 # all implementations except WEBOS_TARGET_MACHINE_IMPL_SIMULATOR
