--- a/orc/orccodemem.c
+++ b/orc/orccodemem.c
@@ -263,6 +263,7 @@ orc_code_region_allocate_codemem (OrcCod
 {
   const char *tmpdir;
 
+/*
   tmpdir = getenv ("TMPDIR");
   if (tmpdir && orc_code_region_allocate_codemem_dual_map (region,
         tmpdir, FALSE)) return;
@@ -277,6 +278,7 @@ orc_code_region_allocate_codemem (OrcCod
   tmpdir = getenv ("HOME");
   if (tmpdir && orc_code_region_allocate_codemem_dual_map (region,
         tmpdir, FALSE)) return;
+*/
 
   if (orc_code_region_allocate_codemem_anon_map (region)) return;
   
