From bce297f97ac42bbae0e0a66000158f33e7e42f18 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Mon, 15 Dec 2014 13:55:26 +0100
Subject: [PATCH] Switch to dbus system bus

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 205030d..04de319 100644
--- a/src/main.c
+++ b/src/main.c
@@ -121,7 +121,7 @@ int main(int argc, char *argv[])
 
 	dbus_error_init(&err);
 
-	conn = g_dbus_setup_bus(DBUS_BUS_SESSION, MMS_SERVICE, &err);
+	conn = g_dbus_setup_bus(DBUS_BUS_SYSTEM, MMS_SERVICE, &err);
 	if (conn == NULL) {
 		if (dbus_error_is_set(&err) == TRUE) {
 			fprintf(stderr, "%s\n", err.message);
-- 
2.1.0

