From 4dc8e0410d99ea719085aea9bbeb57f0dae17d8f Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Fri, 2 Sep 2016 17:56:04 +0000
Subject: [PATCH] Set a transparent background for contextual menus

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/controls/qquickpopupwindow.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/controls/qquickpopupwindow.cpp b/src/controls/qquickpopupwindow.cpp
index 9a6519ed..7ac4851b 100644
--- a/src/controls/qquickpopupwindow.cpp
+++ b/src/controls/qquickpopupwindow.cpp
@@ -52,6 +52,7 @@ QQuickPopupWindow1::QQuickPopupWindow1() :
     m_dismissed(false), m_pressed(false)
 {
     setFlags(Qt::Popup);
+    setColor(QColor(Qt::transparent));
     connect(qApp, SIGNAL(applicationStateChanged(Qt::ApplicationState)),
             this, SLOT(applicationStateChanged(Qt::ApplicationState)));
 }
