From f6ffbe07a8887a08dc67108f73b1f6d0f85620b3 Mon Sep 17 00:00:00 2001
From: Mohammed Sameer <msameer@foolab.org>
Date: Mon, 14 Apr 2014 00:14:49 +0300
Subject: [PATCH 2/6] Downgrade mpeg4videoparse to prevent it from being picked
 up

---
 gst/videoparsers/plugin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/videoparsers/plugin.c b/gst/videoparsers/plugin.c
index 485b0ed..11afe8d 100644
--- a/gst/videoparsers/plugin.c
+++ b/gst/videoparsers/plugin.c
@@ -44,7 +44,7 @@ plugin_init (GstPlugin * plugin)
   ret |= gst_element_register (plugin, "mpegvideoparse",
       GST_RANK_PRIMARY + 1, GST_TYPE_MPEGVIDEO_PARSE);
   ret |= gst_element_register (plugin, "mpeg4videoparse",
-      GST_RANK_PRIMARY + 1, GST_TYPE_MPEG4VIDEO_PARSE);
+      GST_RANK_NONE, GST_TYPE_MPEG4VIDEO_PARSE);
   ret |= gst_element_register (plugin, "pngparse",
       GST_RANK_PRIMARY, GST_TYPE_PNG_PARSE);
   ret |= gst_element_register (plugin, "vc1parse",
-- 
1.8.5.2

