From c3623a824dbd31ec21f02f6c44b17d32f56b3859 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Thu, 14 Sep 2023 12:04:32 +0200
Subject: [PATCH] msgid: Add messages for LuneOS modules

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS specific]

 include/internal/msgid.h | 89 ++++++++++++++++++++++++++++++++++++++--
 1 file changed, 86 insertions(+), 3 deletions(-)

diff --git a/include/internal/msgid.h b/include/internal/msgid.h
index 14df6b4..fd3734e 100644
--- a/include/internal/msgid.h
+++ b/include/internal/msgid.h
@@ -55,18 +55,101 @@
 #define MSGID_NYX_MOD_DEV_INFO_OPEN_ERR                                     "NYXDEV_INFO_OPEN_ERR"
 #define MSGID_NYX_MOD_DEVICEID_OPEN_ERR                                     "NYXDEV_DEVICEID_OPEN_ERR"
 
-/*Display lib open*/
+/** Display lib open*/
 #define MSGID_NYX_MOD_DISP_OPEN_ALREADY_ERR                                 "NYXDIS_OPEN_ALREADY_ERR"
 #define MSGID_NYX_MOD_DISP_OPEN_ERR                                         "NYXDIS_OPEN_ERR"
 #define MSGID_NYX_MOD_DISP_OUT_OF_MEMORY                                    "NYXDIS_OUT_OF_MEM"
 
-/*Security lib open*/
+/** Security lib open*/
 #define MSGID_NYX_MOD_SECU_OPEN_ERR                                         "NYXSEC_OPEN_ERR"
 #define MSGID_NYX_MOD_SECU_OUT_OF_MEMORY                                    "NYXSEC_OUT_OF_MEM"
 
-/*System */
+/** System */
 #define MSGID_NYX_MOD_SYSTEM_OUT_OF_MEMORY                                  "NYXSYS_OUT_OF_MEM"
 #define MSGID_NYX_MOD_SYSTEM_OPEN_ERR                                       "NYXSYS_OPEN_ERR"
 
+/** Mass Storage Mode - MTP */
+#define MSGID_NYX_MOD_MSMMTP_OPEN_ERR                                       "NYXMSM_OPEN_ERR"
+
+/** Ambient Light Sensor */
+#define MSGID_NYX_MOD_ALS_ENABLE_ERR                                        "NYXALS_ENABLE_ERR"
+#define MSGID_NYX_MOD_ALS_DISABLE_ERR                                       "NYXALS_DISABLE_ERR"
+#define MSGID_NYX_MOD_ALS_READ_EVENT_ERR                                    "NYXALS_READ_EVENT_ERR"
+
+/** LED Controller */
+#define MSGID_NYX_MOD_LED_NODEVICE_ERR                                      "NYXLED_NODEVICE_ERR"
+#define MSGID_NYX_MOD_LED_OPENFILE_ERR                                      "NYXLED_OPENFILE_ERR"
+#define MSGID_NYX_MOD_LED_FILE_CONTENT_ERR                                  "NYXLED_FILECONTENT_ERR"
+
+/** Haptics Controller */
+#define MSGID_NYX_MOD_HAPTICS_ODEVICE_FOUND                                 "NYXHAPTICS_DEVICE_FOUND"
+#define MSGID_NYX_MOD_HAPTICS_NODEVICE_ERR                                  "NYXHAPTICS_NODEVICE_ERR"
+#define MSGID_NYX_MOD_HAPTICS_NOSPECIAL_EFF                                 "NYXHAPTICS_NOSPECIAL_EFF"
+#define MSGID_NYX_MOD_HAPTICS_NOPULSES_ERR                                  "NYXHAPTICS_NOPULSES_ERR"
+#define MSGID_NYX_MOD_HAPTICS_TOGGLE_TIMEOUT                                "NYXHAPTICS_TOGGLE_TIMEOUT"
+#define MSGID_NYX_MOD_HAPTICS_VIBRATE_PATTERN                               "NYXHAPTICS_VIBRATE_PATTERN"
+#define MSGID_NYX_MOD_HAPTICS_VIBRATE                                       "NYXHAPTICS_VIBRATE"
+
+/** Keys */
+#define MSGID_NYX_MOD_KEYS_CONF_FILE_ERR                                    "NYXKEYS_CONF_FILE_ERR"
+#define MSGID_NYX_MOD_KEYS_CONF_FILE_PATH_ERR                               "NYXKEYS_CONF_FILE_PATH_ERR"
+#define MSGID_NYX_MOD_KEYS_NEW_INPUT_DEV                                    "NYXKEYS_NEW_INPUT_DEV"
+#define MSGID_NYX_MOD_KEY_EVENT_ERR                                         "NYXKEY_EVENT_ERR"
+#define MSGID_NYX_MOD_KEY_EVENT_READ_ERR                                    "NYXKEY_EVENT_READ_ERR"
+#define MSGID_NYX_MOD_KEYS_OPEN_ERR                                         "NYXKEY_OPEN_ERR"
+#define MSGID_NYX_MOD_KEY_OUT_OF_MEM                                        "NYXKEY_OUT_OF_MEM_ERR"
+
+/** Touchpanel & Touchpanel MTDEV */
+#define MSGID_NYX_MOD_TP_COORDBUF_ERR                                       "NYXTP_COORDBUF_ERR"
+#define MSGID_NYX_MOD_TP_COORDS_ERR                                         "NYXTP_COORDS_ERR"
+#define MSGID_NYX_MOD_TP_NO_FINGER_BUFF                                     "NYXTP_NO_FINGER_BUFF"
+#define MSGID_NYX_MOD_TP_NEW_FINGER                                         "NYXTP_NEW_FINGER"
+#define MSGID_NYX_MOD_TP_FINGER_WT                                          "NYXTP_FINGER_WT"
+#define MSGID_NYX_MOD_TP_FINGER_DOWN                                        "NYXTP_FINGER_DOWN"
+#define MSGID_NYX_MOD_TP_FINGER_UP                                          "NYXTP_FINGER_UP"
+#define MSGID_NYX_MOD_TP_FING_LOW_WT                                        "NYXTP_FING_LOW_WT"
+#define MSGID_NYX_MOD_TP_NOTOUCH_ERR                                        "NYXTP_NOTOUCH_ERR"
+#define MSGID_NYX_MOD_TP_VBOX_OPEN_ERR                                      "NYXTP_VBOX_OPEN_ERR"
+#define MSGID_NYX_MOD_TP_IOCTL_ERR                                          "NYXTP_IOCTL_ERR"
+#define MSGID_NYX_MOD_TP_IOCTL_READ_ERR                                     "NYXTP_IOCTL_READ_ERR"
+#define MSGID_NYX_MOD_TP_IOCTL_REQUEST_ERR                                  "NYXTP_IOCTL_REQUEST_ERR"
+#define MSGID_NYX_MOD_TP_SETPTR_ERR                                         "NYXTP_SETPTR_ERR"
+#define MSGID_NYX_MOD_TP_OPEN_FB_ERR                                        "NYXTP_OPEN_FB_ERR"
+#define MSGID_NYX_MOD_TP_VSCREEN_INFO_ERR                                   "NYXTP_FB_VSCREEN_INFO_ERR"
+#define MSGID_NYX_MOD_TP_OPEN_ERR                                           "NYXTP_OPEN_ERR"
+#define MSGID_NYX_MOD_TP_EVENT_HLIMIT_ERR                                   "NYXTP_EVENT_HLIMIT_ERR"
+#define MSGID_NYX_MOD_TP_EVENT_VLIMIT_ERR                                   "NYXTP_EVENT_VLIMIT_ERR"
+#define MSGID_NYX_MOD_TP_RES_ERR                                            "NYXTP_DP_RES_ERR"
+#define MSGID_NYX_MOD_TP_EVT_READ_ERR                                       "NYXTP_INPUT_EVENT_READ_ERR"
+#define MSGID_NYX_MOD_TP_ABS_ERR                                            "NYXTP_ABS_ERR"
+#define MSGID_NYX_MOD_TP_EVENT_NULL_ERR                                     "NYXTP_EVENT_NULL_ERR"
+#define MSGID_NYX_MOD_TP_INVALID_EVENT                                      "NYXTP_INVALID_EVENT"
+#define MSGID_NYX_MOD_TP_TOOMANY_ITEMS_ERR                                  "NYXTP_TOOMANY_ITEMS_ERR"
+#define MSGID_NYX_MOD_TP_OUT_OF_MEMORY                                      "NYXTP_OUT_OF_MEM_ERR"
+#define MSGID_NYX_MOD_TP_IGNORING_COORD                                     "NYXTP_IGNORING_COORD"
+
+/** Touchpanel MTDEV */
+#define MSGID_NYX_QMUX_TP_COORDBUF_ERR                                      "NYXTP_COORDBUF_ERR"
+#define MSGID_NYX_QMUX_TP_COORDS_ERR                                        "NYXTP_COORDS_ERR"
+#define MSGID_NYX_QMUX_TP_FINGER_WT                                         "NYXTP_FINGER_WT"
+#define MSGID_NYX_QMUX_TP_FING_LOW_WT                                       "NYXTP_FING_LOW_WT"
+#define MSGID_NYX_QMUX_TP_NOTOUCH_ERR                                       "NYXTP_NOTOUCH_ERR"
+#define MSGID_NYX_QMUX_TP_VBOX_OPEN_ERR                                     "NYXTP_VBOX_OPEN_ERR"
+#define MSGID_NYX_QMUX_TP_IOCTL_ERR                                         "NYXTP_IOCTL_ERR"
+#define MSGID_NYX_QMUX_TP_IOCTL_READ_ERR                                    "NYXTP_IOCTL_READ_ERR"
+#define MSGID_NYX_QMUX_TP_IOCTL_REQUEST_ERR                                 "NYXTP_IOCTL_REQUEST_ERR"
+#define MSGID_NYX_QMUX_TP_SETPTR_ERR                                        "NYXTP_SETPTR_ERR"
+#define MSGID_NYX_QMUX_TP_OPEN_FB_ERR                                       "NYXTP_OPEN_FB_ERR"
+#define MSGID_NYX_QMUX_TP_VSCREEN_INFO_ERR                                  "NYXTP_FB_VSCREEN_INFO_ERR"
+#define MSGID_NYX_QMUX_TP_OPEN_ERR                                          "NYXTP_OPEN_ERR"
+#define MSGID_NYX_QMUX_TP_EVENT_HLIMIT_ERR                                  "NYXTP_EVENT_HLIMIT_ERR"
+#define MSGID_NYX_QMUX_TP_EVENT_VLIMIT_ERR                                  "NYXTP_EVENT_VLIMIT_ERR"
+#define MSGID_NYX_QMUX_TP_RES_ERR                                           "NYXTP_DP_RES_ERR"
+#define MSGID_NYX_QMUX_TP_EVT_READ_ERR                                      "NYXTP_INPUT_EVENT_READ_ERR"
+#define MSGID_NYX_QMUX_TP_ABS_ERR                                           "NYXTP_ABS_ERR"
+#define MSGID_NYX_QMUX_TP_EVENT_NULL_ERR                                    "NYXTP_EVENT_NULL_ERR"
+#define MSGID_NYX_QMUX_TP_INVALID_EVENT                                     "NYXTP_INVALID_EVENT"
+#define MSGID_NYX_QMUX_TP_TOOMANY_ITEMS_ERR                                 "NYXTP_TOOMANY_ITEMS_ERR"
+#define MSGID_NYX_QMUX_TP_OUT_OF_MEMORY                                     "NYXTP_OUT_OF_MEM_ERR"
 
 #endif // __NYX__MODULES__MESSAGE__LOG_H__
