From b29908ac6e8c76a8d9baeae7675632283ba79a80 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 7 Jul 2014 13:02:55 +0200
Subject: [PATCH] droid-extcon.c: adapt to new pulseaudio 5 APIs

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/droid/droid-extcon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/droid/droid-extcon.c b/src/droid/droid-extcon.c
index 088afb3..697957d 100644
--- a/src/droid/droid-extcon.c
+++ b/src/droid/droid-extcon.c
@@ -122,13 +122,13 @@ static void notify_ports(pa_droid_extcon *u, struct droid_switch *as) {
     pa_log_debug("Value of switch %s is now %d.", as->name, as->current_value);
 
     PA_HASHMAP_FOREACH(p, u->card->ports, state) {
-        if (p->is_output) {
+        if (p->direction == PA_DIRECTION_OUTPUT) {
             if (!strcmp(p->name, "output-wired_headset"))
                 pa_device_port_set_available(p, hsmic_avail(as->current_value));
             if (!strcmp(p->name, "output-wired_headphone"))
                 pa_device_port_set_available(p, hponly_avail(as->current_value));
         }
-        if (p->is_input) {
+        if (p->direction == PA_DIRECTION_INPUT) {
             if (!strcmp(p->name, "input-wired_headset"))
                 pa_device_port_set_available(p, hsmic_avail(as->current_value));
         }
