From d70af967656e1b415fc0f823b2697f1d851c4b8e Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Wed, 7 Oct 2015 11:28:43 +0200
Subject: [PATCH] Remove AUDIO_DEVICE_BIT_IN for mako again

To see if this fixes the issues on Mako with PA not working anymore after upgrade.
---
 src/common/droid-util.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/common/droid-util.c b/src/common/droid-util.c
index 6c5cb9a..e1d0cf9 100644
--- a/src/common/droid-util.c
+++ b/src/common/droid-util.c
@@ -1148,7 +1148,9 @@ static void add_i_ports(pa_droid_mapping *am) {
         if (devices & cur_device) {
 
 #if DROID_HAL >= 2
+#ifndef DROID_DEVICE_MAKO
             cur_device |= AUDIO_DEVICE_BIT_IN;
+#endif            
 #endif
 
             pa_assert_se(pa_droid_input_port_name(cur_device, &name));
