From a60e112d04eeb1b2c1ed32c1644ff1f7b5f5a90f Mon Sep 17 00:00:00 2001
From: @ <@9Œä×U>
Date: Sat, 26 Nov 2016 17:22:49 +0000
Subject: [PATCH] Don't include QCOM_HARDWARE related defines

---
 src/common/droid-util-51.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/droid-util-51.h b/src/common/droid-util-51.h
index d64f1a1..884d67e 100644
--- a/src/common/droid-util-51.h
+++ b/src/common/droid-util-51.h
@@ -30,7 +30,7 @@
 // device will need to have an exception below (just like i9305).
 // TODO: things elegantly
 #if defined(__arm__) && !defined(DROID_DEVICE_I9305)
-#define QCOM_HARDWARE
+// #define QCOM_HARDWARE
 #endif
 
 #ifdef QCOM_BSP
--
libgit2 0.24.0

