From c37ecf636243b2be11192a3a30f3165e20fb52ad Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Wed, 13 Jan 2016 20:27:15 +0100
Subject: [PATCH 02/10] Add alarms.xml to CMakeLists.txt

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b59e99..3961bf8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,3 +122,4 @@ webos_build_system_bus_files()
 webos_config_build_doxygen(doc Doxyfile)
 
 install(FILES files/conf/sleepd.conf DESTINATION ${WEBOS_INSTALL_DEFAULTCONFDIR})
+install(FILES files/conf/alarms.xml DESTINATION ${WEBOS_INSTALL_PREFERENCESDIR}/com.palm.sleep)
-- 
2.41.0.windows.3

