From 60293250b606e384193e1b12775393ac8f4d16a2 Mon Sep 17 00:00:00 2001
From: Mohammed Hassan <mohammed.hassan@jolla.com>
Date: Thu, 30 Oct 2014 15:06:47 +0200
Subject: [PATCH 5/5] Downgrade mpeg4videoparse to prevent it from being picked
 up

---
 gst/videoparsers/plugin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/videoparsers/plugin.c b/gst/videoparsers/plugin.c
index 79d1df6..bac4141 100644
--- a/gst/videoparsers/plugin.c
+++ b/gst/videoparsers/plugin.c
@@ -45,7 +45,7 @@ plugin_init (GstPlugin * plugin)
   ret |= gst_element_register (plugin, "mpegvideoparse",
       GST_RANK_PRIMARY + 1, GST_TYPE_MPEGVIDEO_PARSE);
   ret |= gst_element_register (plugin, "mpeg4videoparse",
-      GST_RANK_PRIMARY + 1, GST_TYPE_MPEG4VIDEO_PARSE);
+      GST_RANK_NONE, GST_TYPE_MPEG4VIDEO_PARSE);
   ret |= gst_element_register (plugin, "pngparse",
       GST_RANK_PRIMARY, GST_TYPE_PNG_PARSE);
   ret |= gst_element_register (plugin, "h265parse",
-- 
1.8.4.rc3

