diff -r 56e9c709db7e src/video/wayland/SDL_waylandevents.c
--- a/src/video/wayland/SDL_waylandevents.c	Wed Feb 12 18:12:14 2014 -0300
+++ b/src/video/wayland/SDL_waylandevents.c	Thu Feb 13 12:29:50 2014 +0100
@@ -309,6 +309,9 @@
 {
     struct SDL_WaylandInput *input = data;
 
+    if (!input->xkb.state)
+        return;
+
     WAYLAND_xkb_state_update_mask(input->xkb.state, mods_depressed, mods_latched,
                           mods_locked, 0, 0, group);
 }
