diff --git a/run-js-service b/run-js-service
index 775fe6e..eef3653 100644
--- a/run-js-service
+++ b/run-js-service
@@ -31,7 +31,7 @@ fi
 
 # Jail master switch overrides the other jail switches- if
 # this is off, all js jails are off.  Valid values are "on" and "off".
-jail=on
+jail=off
 # Jail palm services.  Valid values are "off", "collective" and "individual"
 # Collective means that all palm services share the same jail.
 # so one could read data owned by another, but startup of new
