From 792f1b3618cd5f9ac4e0ef3c42e196e40cd81c08 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Wed, 1 May 2013 17:53:05 +0200
Subject: [PATCH] Don't treat none as security type

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/wifi_service.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/wifi_service.c b/src/wifi_service.c
index a5714e4..5f23441 100644
--- a/src/wifi_service.c
+++ b/src/wifi_service.c
@@ -350,6 +350,12 @@ static void add_service(connman_service_t *service, jvalue_ref *network)
 			// We do not support enterprise security i.e "ieee8021x" security type
 			if(!g_strcmp0(service->security[i],"ieee8021x"))
 				supported = FALSE;
+
+			// We don't want "none" listed as security type as a empty list
+			// already means "none".
+			if (!g_strcmp0(service->security[i], "none"))
+				continue;
+
 			jarray_append(security_list, jstring_create(service->security[i]));
 		}
 		jobject_put(*network, J_CSTR_TO_JVAL("availableSecurityTypes"),security_list);
-- 
1.8.1.2

