From 76cc3dd0df9d9a92587d54850fcbe37a40c34933 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Mon, 16 Feb 2015 19:53:32 +0100
Subject: [PATCH 6/6] Add networkConfidenceLevel field required by
 activitymanager

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/connectionmanager_service.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/connectionmanager_service.c b/src/connectionmanager_service.c
index f9976e4..223a73c 100644
--- a/src/connectionmanager_service.c
+++ b/src/connectionmanager_service.c
@@ -104,9 +104,14 @@ static void update_connection_status(connman_service_t *connected_service, jvalu
 			if(NULL != connected_service->name)
 				jobject_put(*status, J_CSTR_TO_JVAL("ssid"), jstring_create(connected_service->name));
 			jobject_put(*status, J_CSTR_TO_JVAL("isWakeOnWifiEnabled"), jboolean_create(false));
+			jobject_put(*status, J_CSTR_TO_JVAL("signalLevel"), jnumber_create_i32(connected_service->strength));
 		}
 		const char *s = (connman_state == CONNMAN_SERVICE_STATE_ONLINE)?"yes":"no";
 		jobject_put(*status, J_CSTR_TO_JVAL("onInternet"), jstring_create(s));
+
+		/* unless we have something to determine the confidence level we always
+		 * provide execellent as proper default. */
+		jobject_put(*status, J_CSTR_TO_JVAL("networkConfidenceLevel"), jstring_create("excellent"));
 	}
 	else
 		jobject_put(*status, J_CSTR_TO_JVAL("state"), jstring_create("disconnected"));
-- 
2.1.0

