From 2d0caf759248fa02eb685c13156778af30877242 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 2 Dec 2023 12:48:52 +0000
Subject: [PATCH] LSMessageIsSubscription: test if payload is a JSON object
 first

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 src/libluna-service2/message.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libluna-service2/message.c b/src/libluna-service2/message.c
index b39393a..3440626 100644
--- a/src/libluna-service2/message.c
+++ b/src/libluna-service2/message.c
@@ -518,7 +518,7 @@ LSMessageIsSubscription(LSMessage *message)
 
     jvalue_ref object = jdom_parse(j_cstr_to_buffer(payload), DOMOPT_NOOPT,
                                    &schemaInfo);
-    if (jis_null(object))
+    if (jis_null(object) || !jis_object(object))
         goto exit;
 
     if (!jobject_get_exists(object, J_CSTR_TO_BUF("subscribe"),
-- 
2.34.1

