From fa52e3c2a49de7b191d078439be5241525a2cad2 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 26 Sep 2015 22:28:22 +0200
Subject: [PATCH 01/14] Add PalmSystemBridge feature

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/config/embedded_linux.pri | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/config/embedded_linux.pri b/src/core/config/embedded_linux.pri
index 7a909f1..17fece1 100644
--- a/src/core/config/embedded_linux.pri
+++ b/src/core/config/embedded_linux.pri
@@ -40,5 +40,6 @@ GYP_CONFIG += \
     v8_use_snapshot=false \
     want_separate_host_toolset=1 \
     angle_enable_gl=0 \
+    enable_palmbridge=1 \
 
 WEBENGINE_CONFIG *= reduce_binary_size
-- 
2.7.4

