From 341f5775f903b77cc945bbe38a4a2fd92453037a Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 15 Aug 2016 09:26:04 +0000
Subject: [PATCH] Enable password echo

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/web_engine_settings.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/web_engine_settings.cpp b/src/core/web_engine_settings.cpp
index b6250a65..28ba1799 100644
--- a/src/core/web_engine_settings.cpp
+++ b/src/core/web_engine_settings.cpp
@@ -349,6 +349,7 @@ void WebEngineSettings::applySettingsToWebPreferences(content::WebPreferences *p
     prefs->available_hover_types = ui::HOVER_TYPE_HOVER;
     prefs->primary_hover_type = ui::HOVER_TYPE_HOVER;
 #endif
+    prefs->password_echo_enabled = true;
     if (prefs->viewport_enabled) {
         // We need to enable the viewport options together as it doesn't really work
         // to enable them separately. With viewport-enabled we match Android defaults.
