From a6b3e762f40473be835eeb875446bc9eb07a80be Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 15 Aug 2016 09:26:04 +0000
Subject: [PATCH] Enable password echo

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/web_engine_settings.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/web_engine_settings.cpp b/src/core/web_engine_settings.cpp
index eeb8ffc..8c3fc54 100644
--- a/src/core/web_engine_settings.cpp
+++ b/src/core/web_engine_settings.cpp
@@ -341,6 +341,7 @@ void WebEngineSettings::applySettingsToWebPreferences(content::WebPreferences *p
     // Override for now
     prefs->touch_enabled = isTouchEventsAPIEnabled();
     prefs->device_supports_touch = isTouchScreenAvailable();
+    prefs->password_echo_enabled = true;
     if (prefs->viewport_enabled) {
         // We need to enable the viewport options together as it doesn't really work
         // to enable them separately. With viewport-enabled we match Android defaults.
