From e2a3b4a29b429ef3264bab021a1b0eedd3b88fa5 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 15 Aug 2016 09:26:04 +0000
Subject: [PATCH] Enable password echo

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/web_engine_settings.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/web_engine_settings.cpp b/src/core/web_engine_settings.cpp
index 191eaa6f..749b5707 100644
--- a/src/core/web_engine_settings.cpp
+++ b/src/core/web_engine_settings.cpp
@@ -372,6 +372,7 @@ void WebEngineSettings::applySettingsToWebPreferences(content::WebPreferences *p
 {
     // Override for now
     prefs->touch_event_feature_detection_enabled = isTouchEventsAPIEnabled();
+    prefs->password_echo_enabled = true;
     if (prefs->viewport_enabled) {
         // We need to enable the viewport options together as it doesn't really work
         // to enable them separately. With viewport-enabled we match Android defaults.
