From 4b95efa1aaf55f3123000addb52963630a35194e Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 15 Aug 2016 09:26:04 +0000
Subject: [PATCH 12/18] Enable password echo

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/web_engine_settings.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/web_engine_settings.cpp b/src/core/web_engine_settings.cpp
index 7d7af2d..a163fb5 100644
--- a/src/core/web_engine_settings.cpp
+++ b/src/core/web_engine_settings.cpp
@@ -341,6 +341,7 @@ void WebEngineSettings::applySettingsToWebPreferences(content::WebPreferences *p
     // Override for now
     prefs->touch_enabled = isTouchEventsAPIEnabled();
     prefs->device_supports_touch = isTouchScreenAvailable();
+    prefs->password_echo_enabled = true;
     if (prefs->viewport_enabled) {
         // We should enable viewport and viewport-meta together, but since 5.7 we
         // no longer have a command-line flag for viewport-meta.
-- 
2.7.4

