From aeb1dbc59a169c1ad548f377958bcac63aa08f8c Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 15 Aug 2016 09:26:04 +0000
Subject: [PATCH 13/14] Enable password echo

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/web_engine_settings.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/web_engine_settings.cpp b/src/core/web_engine_settings.cpp
index 83336bd..d13b7af 100644
--- a/src/core/web_engine_settings.cpp
+++ b/src/core/web_engine_settings.cpp
@@ -315,6 +315,7 @@ void WebEngineSettings::applySettingsToWebPreferences(content::WebPreferences *p
 {
     // Override for now
     prefs->touch_enabled = isTouchScreenAvailable();
+    prefs->password_echo_enabled = true;
 
     // Attributes mapping.
     prefs->loads_images_automatically = testAttribute(AutoLoadImages);
-- 
2.7.4

