From 19b2557e2938933b7c887dd7657668549c4fd4a1 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 15 Aug 2016 09:26:04 +0000
Subject: [PATCH 13/14] Enable password echo

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/web_engine_settings.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/web_engine_settings.cpp b/src/core/web_engine_settings.cpp
index aef5a20..b042a25 100644
--- a/src/core/web_engine_settings.cpp
+++ b/src/core/web_engine_settings.cpp
@@ -315,6 +315,8 @@ void WebEngineSettings::applySettingsToWebPreferences(content::WebPreferences *p
 {
     // Override for now
     prefs->touch_enabled = isTouchScreenAvailable();
+    prefs->password_echo_enabled = true;
+

     // Attributes mapping.
     prefs->loads_images_automatically = testAttribute(AutoLoadImages);
-- 
2.7.4

