diff --git a/src/core/render_widget_host_view_qt.cpp b/src/core/render_widget_host_view_qt.cpp
index 5105214..fea7313 100644
--- a/src/core/render_widget_host_view_qt.cpp
+++ b/src/core/render_widget_host_view_qt.cpp
@@ -43,6 +43,7 @@
 #include "browser_accessibility_manager_qt.h"
 #include "browser_accessibility_qt.h"
 #include "chromium_overrides.h"
+#include "web_contents_view_qt.h"
 #include "delegated_frame_node.h"
 #include "qtwebenginecoreglobal_p.h"
 #include "render_widget_host_view_qt_delegate.h"
@@ -715,7 +716,7 @@ void RenderWidgetHostViewQt::GetScreenInfo(content::ScreenInfo* results)
         GetScreenInfoFromNativeWindow(window, results);
     }
     else {
-        GetDefaultScreenInfo(results);
+        content::WebContentsView::GetDefaultScreenInfo(results);
     }
 
     // Support experimental.viewport.devicePixelRatio
