diff -uNr lxc-2.0.5.orig/src/lxc/bdev/lxclvm.c lxc-2.0.5/src/lxc/bdev/lxclvm.c
--- lxc-2.0.5.orig/src/lxc/bdev/lxclvm.c	2016-10-05 11:47:41.000000000 +0200
+++ lxc-2.0.5/src/lxc/bdev/lxclvm.c	2017-02-03 19:45:06.461468704 +0100
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 
 #include "bdev.h"
 #include "log.h"
diff -uNr lxc-2.0.5.orig/src/lxc/conf.c lxc-2.0.5/src/lxc/conf.c
--- lxc-2.0.5.orig/src/lxc/conf.c	2016-10-05 11:47:41.000000000 +0200
+++ lxc-2.0.5/src/lxc/conf.c	2017-02-03 19:45:09.990469005 +0100
@@ -35,6 +35,7 @@
 #include <sys/wait.h>
 #include <sys/syscall.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <pwd.h>
 #include <grp.h>
 #include <time.h>
diff -uNr lxc-2.0.5.orig/src/lxc/lxccontainer.c lxc-2.0.5/src/lxc/lxccontainer.c
--- lxc-2.0.5.orig/src/lxc/lxccontainer.c	2016-10-05 11:47:41.000000000 +0200
+++ lxc-2.0.5/src/lxc/lxccontainer.c	2017-02-03 19:44:57.093467907 +0100
@@ -36,6 +36,7 @@
 #include <sys/mount.h>
 #include <sys/syscall.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/wait.h>
 
 #include "attach.h"
