diff --git a/maliit-keyboard/plugin/inputmethod.cpp b/maliit-keyboard/plugin/inputmethod.cpp
index 9a0e187..10b4d91 100644
--- a/maliit-keyboard/plugin/inputmethod.cpp
+++ b/maliit-keyboard/plugin/inputmethod.cpp
@@ -446,7 +448,9 @@ QString InputMethod::activeSubView(Maliit::HandlerState state) const
 void InputMethod::handleAppOrientationChanged(int angle)
 {
     Q_D(InputMethod);
-    d->setLayoutOrientation((angle == 0 || angle == 180)
+
+    const QSize &screen_size(QGuiApplication::primaryScreen()->availableSize());
+    d->setLayoutOrientation(( (angle == 0 || angle == 180) && (screen_size.width() >= screen_size.height()) )
                             ? Logic::LayoutHelper::Landscape
                             : Logic::LayoutHelper::Portrait);
 }
