From 50dbb8209bf65e4d24ccd06d3ca05004d0ddc63c Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@freescale.com>
Date: Mon, 24 Mar 2014 12:43:03 -0400
Subject: [PATCH] lxc-busybox: follow symlinks when inspecting busybox binary

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@freescale.com>
Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 templates/lxc-busybox.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/templates/lxc-busybox.in b/templates/lxc-busybox.in
index e5a512a..246e743 100644
--- a/templates/lxc-busybox.in
+++ b/templates/lxc-busybox.in
@@ -187,7 +187,7 @@ configure_busybox()
         return 1
     fi
 
-    file $(which busybox) | grep -q "statically linked"
+    file -L $(which busybox) | grep -q "statically linked"
     if [ $? -ne 0 ]; then
         echo "warning : busybox is not statically linked."
         echo "warning : The template script may not correctly"
-- 
1.9.rc1

