From 6da7ef9b92d97b16d0c065d95de2e60dcb255baf Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Wed, 12 Aug 2026 22:11:59 +0200
Subject: [PATCH 04/17] DefaultSettings.qml: Use Prelude for LuneOS

To give a more uniform experience in LuneOS.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

Upstream-Status: Pending
---
 base/qml/WebOSCompositorBase/DefaultSettings.qml | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/base/qml/WebOSCompositorBase/DefaultSettings.qml b/base/qml/WebOSCompositorBase/DefaultSettings.qml
index bed0f10..faaa074 100644
--- a/base/qml/WebOSCompositorBase/DefaultSettings.qml
+++ b/base/qml/WebOSCompositorBase/DefaultSettings.qml
@@ -91,7 +91,7 @@ QtObject {
                 "height": 85,
                 "backgroundColor": "#ffffff",
                 "activatedColor": "#cf0652",
-                "fontName": "Miso",
+                "fontName": "Prelude",
                 "fontSize": 52,
                 "textColor": "#4b4b4b",
                 "textActivatedColor": "#ffffff"
@@ -108,8 +108,8 @@ QtObject {
                 "buttonMargins": 40,
                 "foregroundColor": "#ffffff",
                 "backgroundColor": "#4d4d4d",
-                "headerFont": "Miso",
-                "fontName": "Museo Sans",
+                "headerFont": "Prelude",
+                "fontName": "Prelude",
                 "fontSize": 82,
                 "fontWeight": Font.Bold,
                 "descriptionFontSize": 26,
@@ -130,7 +130,7 @@ QtObject {
                 "backgroundColor": "#4d4d4d",
                 "activatedColor": "#cf0652",
                 "iconSize": 80,
-                "fontName": "Museo Sans",
+                "fontName": "Prelude",
                 "fontSize": 26,
                 "fontWeight": Font.Bold,
                 "textLineHeight": 40,
@@ -143,7 +143,7 @@ QtObject {
                 "height": 320,
                 "backgroundColor": "#4d4d4d",
                 "titleColor": "#d0d0d0",
-                "titleFontFamilyName": "Miso",
+                "titleFontFamilyName": "Prelude",
                 "titleFontSize": 73,
                 "titleTextHeight": 86,
                 "titleTextLeftMargin": 36,
@@ -172,7 +172,7 @@ QtObject {
                 "numPadNormalFontColor": "#eeeeee",
                 "numPadFontActiveColor": "#ffffff",
                 "numPadFontDisabledColor": "#8b8b8b",
-                "numPadFontFamilyName": "Museo Sans",
+                "numPadFontFamilyName": "Prelude",
                 "numPadFontSize": 28,
                 "slideAnimationDuration": 500
             }
