From 2eb2d81beab8f69510bf1d4408b5dda1a196c681 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 11 Mar 2023 12:52:39 +0200
Subject: [PATCH] qmldir: expose NotificationService component

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 base/qml/WebOSCompositorBase/qmldir | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/qml/WebOSCompositorBase/qmldir b/base/qml/WebOSCompositorBase/qmldir
index 6846e4b..bff81bf 100644
--- a/base/qml/WebOSCompositorBase/qmldir
+++ b/base/qml/WebOSCompositorBase/qmldir
@@ -25,6 +25,7 @@ FullscreenView 1.0 views/FullscreenView.qml
 KeyboardView 1.0 views/KeyboardView.qml
 Launcher 1.0 views/Launcher.qml
 NotificationView 1.0 views/NotificationView.qml
+NotificationService 1.0 views/notification/NotificationService.qml
 OverlayView 1.0 views/OverlayView.qml
 PopupView 1.0 views/PopupView.qml
 Spinner 1.0 views/Spinner.qml
