From 356df47f083f465f6b5038225f2e6255855ad21b Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Wed, 12 Aug 2026 22:12:00 +0200
Subject: [PATCH 10/17] qmldir: expose NotificationService component

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 base/qml/WebOSCompositorBase/qmldir | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/qml/WebOSCompositorBase/qmldir b/base/qml/WebOSCompositorBase/qmldir
index 4fee381..db9dd96 100644
--- a/base/qml/WebOSCompositorBase/qmldir
+++ b/base/qml/WebOSCompositorBase/qmldir
@@ -26,6 +26,7 @@ FullscreenView 1.0 views/FullscreenView.qml
 KeyboardView 1.0 views/KeyboardView.qml
 Launcher 1.0 views/Launcher.qml
 NotificationView 1.0 views/NotificationView.qml
+NotificationService 1.0 views/notification/NotificationService.qml
 OverlayView 1.0 views/OverlayView.qml
 PopupView 1.0 views/PopupView.qml
 Spinner 1.0 views/Spinner.qml
