From 7f7e8fbffead67c3a958a9eb90cd95922166121f Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 23 May 2024 17:43:13 +0000
Subject: [PATCH] Wait for graphic card before LSM startup

Upstream-Status: Pending

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 base/startup/surface-manager.sh.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/base/startup/surface-manager.sh.in b/base/startup/surface-manager.sh.in
index 961f66d..b71b285 100755
--- a/base/startup/surface-manager.sh.in
+++ b/base/startup/surface-manager.sh.in
@@ -48,5 +48,10 @@ else
     pmlog warning "Missing ${PRODUCT_ENVIRONMENT}. Use default values."
 fi
 
+# We need the graphic card to be detected first, otherwise surface-manager won't launch
+while [ ! -e /dev/dri/card0 ] && [ ! -e /dev/graphics/fb0 ]; do
+    sleep 1
+done
+
 pmlog info "Starting @WEBOS_INSTALL_BINS@/surface-manager with flags: -platform ${WEBOS_COMPOSITOR_PLATFORM}, compositor extra options: ${WEBOS_COMPOSITOR_EXTRA_OPTIONS}"
 exec @WEBOS_INSTALL_BINS@/surface-manager -platform ${WEBOS_COMPOSITOR_PLATFORM} ${WEBOS_COMPOSITOR_EXTRA_OPTIONS}
