From f55eedd3329ab66e77982f48aad9a5c3fe6dc043 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 6 Feb 2022 18:25:47 +0000
Subject: [PATCH 3/5] AppInstaller: rescan apps after install

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 src/installer/AppInstaller.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/installer/AppInstaller.cpp b/src/installer/AppInstaller.cpp
index df9b1ae..1a9d84e 100644
--- a/src/installer/AppInstaller.cpp
+++ b/src/installer/AppInstaller.cpp
@@ -23,6 +23,7 @@
 #include "base/JUtil.h"
 #include "base/Utils.h"
 #include "base/Logging.h"
+#include "util/Logger.h"
 #include "installer/AppInstallerErrors.h"
 #include "installer/Task.h"
 #include "settings/Settings.h"
@@ -182,6 +183,13 @@ void AppInstaller::onFinishTask(const Task &task)
             });
         }
     }
+
+    std::string errorText;
+    LSCaller caller = LSUtils::acquireCaller("com.webos.appInstallService");
+    if (!caller.Call("luna://com.palm.applicationManager/rescan", "{}", nullptr, this, nullptr, errorText)) {
+        Logger::error("AppInstaller", __FUNCTION__, "rescan error: " + errorText);
+    }
+    
     //resumeAllTask(Task::SYSTEM); //TODO : need to check
 }
 
-- 
2.37.3.windows.1

