From 22eed699b66934eb775a4714446361ce86c9a9b3 Mon Sep 17 00:00:00 2001
From: Isaac Garzon <isagar2004@gmail.com>
Date: Thu, 15 Oct 2015 00:29:45 +0300
Subject: [PATCH] Added password mask delay

---
 src/client/qwaylandintegration.cpp | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/src/client/qwaylandintegration.cpp b/src/client/qwaylandintegration.cpp
index b804988b..dcf44cd3 100644
--- a/src/client/qwaylandintegration.cpp
+++ b/src/client/qwaylandintegration.cpp
@@ -88,6 +88,22 @@ QT_BEGIN_NAMESPACE
 
 namespace QtWaylandClient {
 
+class WaylandLunaTheme: public QGenericUnixTheme
+{
+public:
+    virtual QVariant themeHint(ThemeHint hint) const
+    {
+        switch (hint) {
+        case QPlatformTheme::PasswordMaskDelay:
+            return int(1000);
+        case QPlatformTheme::PasswordMaskCharacter:
+            return QChar(0x2022);
+        default:
+            return QGenericUnixTheme::themeHint(hint);
+        }
+    }
+};
+
 class GenericWaylandTheme: public QGenericUnixTheme
 {
 public:
@@ -287,11 +303,13 @@ QWaylandDisplay *QWaylandIntegration::display() const
 
 QStringList QWaylandIntegration::themeNames() const
 {
-    return GenericWaylandTheme::themeNames();
+    return QStringList("wayland_luna");
 }
 
 QPlatformTheme *QWaylandIntegration::createPlatformTheme(const QString &name) const
 {
+    if (name == "wayland_luna")
+        return new WaylandLunaTheme;
     return GenericWaylandTheme::createUnixTheme(name);
 }
 
