From 4a3c26096c0ec3d46f53e319179d9230ce3308b5 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Fri, 28 Jul 2017 15:19:32 +0000
Subject: [PATCH] QWaylandWindow: reset window should reset mask

---
 src/client/qwaylandwindow.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index b6f16f0..fafd1c6 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -230,6 +230,8 @@ void QWaylandWindow::reset()
 
     if (mFrameCallback)
         wl_callback_destroy(mFrameCallback);
+        
+    mMask = QRegion();
 }
 
 QWaylandWindow *QWaylandWindow::fromWlSurface(::wl_surface *surface)
-- 
2.7.4

