From 516b20f51358c6dcb4637230eec1d5ab64331f17 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 6 Jul 2025 15:16:18 +0200
Subject: [PATCH] HfpHFRole: init mHfpOfonoManager to nullptr

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Upstream-Status: Pending
---
 src/HF/hfphfrole.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/HF/hfphfrole.cpp b/src/HF/hfphfrole.cpp
index d3bd000..70e5f99 100644
--- a/src/HF/hfphfrole.cpp
+++ b/src/HF/hfphfrole.cpp
@@ -35,6 +35,7 @@ HfpHFRole::HfpHFRole(BluetoothHfpService *service) :
         mGetStatusSubscription(nullptr),
 	mHFLS2Call(nullptr),
 	mHFDevice(nullptr),
+	mHfpOfonoManager(nullptr),
 	mNameWatch(G_BUS_TYPE_SYSTEM, "org.ofono")
 {
 	LS_CREATE_CATEGORY_BEGIN(HfpHFRole, adapter)
