From 0fb8d768145447b9eadd9064024d2e5bd847a7f7 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Fri, 22 Jul 2022 12:15:58 +0200
Subject: [PATCH] DefaultSettings.qml: Use Prelude for LuneOS

To give a more uniform experience in LuneOS.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 base/qml/WebOSCompositorBase/DefaultSettings.qml | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/base/qml/WebOSCompositorBase/DefaultSettings.qml b/base/qml/WebOSCompositorBase/DefaultSettings.qml
index 9c7c082..1ccf0e1 100644
--- a/base/qml/WebOSCompositorBase/DefaultSettings.qml
+++ b/base/qml/WebOSCompositorBase/DefaultSettings.qml
@@ -88,7 +88,7 @@ QtObject {
                 "height": 85,
                 "backgroundColor": "#ffffff",
                 "activatedColor": "#cf0652",
-                "fontName": "Miso",
+                "fontName": "Prelude",
                 "fontSize": 52,
                 "textColor": "#4b4b4b",
                 "textActivatedColor": "#ffffff"
@@ -105,8 +105,8 @@ QtObject {
                 "buttonMargins": 40,
                 "foregroundColor": "#ffffff",
                 "backgroundColor": "#4d4d4d",
-                "headerFont": "Miso",
-                "fontName": "Museo Sans",
+                "headerFont": "Prelude",
+                "fontName": "Prelude",
                 "fontSize": 82,
                 "fontWeight": Font.Bold,
                 "descriptionFontSize": 26,
@@ -127,7 +127,7 @@ QtObject {
                 "backgroundColor": "#4d4d4d",
                 "activatedColor": "#cf0652",
                 "iconSize": 80,
-                "fontName": "Museo Sans",
+                "fontName": "Prelude",
                 "fontSize": 26,
                 "fontWeight": Font.Bold,
                 "textLineHeight": 40,
@@ -140,7 +140,7 @@ QtObject {
                 "height": 320,
                 "backgroundColor": "#4d4d4d",
                 "titleColor": "#d0d0d0",
-                "titleFontFamilyName": "Miso",
+                "titleFontFamilyName": "Prelude",
                 "titleFontSize": 73,
                 "titleTextHeight": 86,
                 "titleTextLeftMargin": 36,
@@ -169,7 +169,7 @@ QtObject {
                 "numPadNormalFontColor": "#eeeeee",
                 "numPadFontActiveColor": "#ffffff",
                 "numPadFontDisabledColor": "#8b8b8b",
-                "numPadFontFamilyName": "Museo Sans",
+                "numPadFontFamilyName": "Prelude",
                 "numPadFontSize": 28,
                 "slideAnimationDuration": 500
             }
-- 
2.31.0.windows.1

