From 84fb8fd3f430894870c917525cd963a8b781199b Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 4 Jul 2014 12:12:19 +0200
Subject: [PATCH] binding.gyp: Use pkg-config to find taglib flags

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 binding.gyp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/binding.gyp b/binding.gyp
index 3211dd9..00440a8 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -1,9 +1,12 @@
 {
+  "variables": {
+    'pkg-config': 'pkg-config' 
+  },
   "targets": [
     {
       "target_name": "taglib",
       "sources": ["src/bufferstream.c", "src/tag.cc", "src/taglib.cc"],
-      "libraries": ["<!(taglib-config --libs)"],
+      "libraries": ["<!@(<(pkg-config) --libs-only-l taglib)"],
       'conditions': [
         ['OS=="mac"', {
           # cflags on OS X are stupid and have to be defined like this
@@ -15,7 +18,7 @@
           }
         }, {
           'cflags': [
-            '<!@(taglib-config --cflags)'
+            '<!@(<(pkg-config) --cflags taglib)'
           ],
         }]
       ]
-- 
2.0.0

