From c84b8720909891ec7e2fdd49cfbee3aa3e3c4974 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Wed, 16 Nov 2022 18:01:06 +0000
Subject: [PATCH] palmGetResource: improve legacy compatibility

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
Upstream-Status: Pending

 .../renderer/resources/webossystem_injection.js | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/src/neva/injection/renderer/resources/webossystem_injection.js b/src/neva/injection/renderer/resources/webossystem_injection.js
index 004f22c0c5..a4add503e8 100644
--- a/src/neva/injection/renderer/resources/webossystem_injection.js
+++ b/src/neva/injection/renderer/resources/webossystem_injection.js
@@ -15,7 +15,22 @@
 // SPDX-License-Identifier: Apache-2.0
 
 window.webOSGetResource = function(p1, p2) {
-  return webOSSystem.getResource(p1, p2);
+  var a = p1 || "";
+  var b = p2 || "";
+
+  if(a.indexOf("file://") === 0){
+      a = a.substring(7);
+  }
+
+  var result = webOSSystem.getResource(a, b);
+
+  if (result.length === 0)
+    return "";
+
+  if (b === "const json")
+      return JSON.parse(result);
+
+  return result;
 };
 
 window.webos = {
