From 9b9243d8e4beeb7c5ba534e206fc401aea92476c Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Fri, 5 Jul 2019 08:29:20 +0200
Subject: [PATCH] Don't remove idle check from mainloop when currently
 suspended

Signed-off-by: Simon Busch <morphis@gravedo.de>
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/pwrevents/suspend.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pwrevents/suspend.c b/src/pwrevents/suspend.c
index 649abf1..4b1bb90 100644
--- a/src/pwrevents/suspend.c
+++ b/src/pwrevents/suspend.c
@@ -248,7 +248,7 @@ ScheduleIdleCheck(int interval_ms, bool fromPoll)
 {
     if (idle_scheduler)
     {
-        SLEEPDLOG_DEBUG("Scheduling new ilde check in %d ms", interval_ms);
+        SLEEPDLOG_DEBUG("Scheduling new idle check in %d ms", interval_ms);
         g_timer_source_set_interval(idle_scheduler, interval_ms, fromPoll);
     }
     else
@@ -282,7 +282,7 @@ IdleCheck(gpointer ctx)
 
     if (gCurrentStateNode.state == kPowerStateKernelResume) {
         SLEEPDLOG_DEBUG("Not rescheduling idle check cause we're in sleep mode");
-        return FALSE;
+        return TRUE;
     }
 
     SLEEPDLOG_DEBUG("IdleCheck: state %s", StateToStr(gCurrentStateNode.state));
