From 28a95c8c1dd1f433d2dd64317e15698ab10f366e Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 1 Apr 2019 13:04:20 +0000
Subject: [PATCH] ashmem, binder: add modules_install target to Makefiles

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Upstream-Status: Pending
---
 ashmem/Makefile | 3 +++
 binder/Makefile | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/ashmem/Makefile b/ashmem/Makefile
index 017e5b7..fddaa31 100644
--- a/ashmem/Makefile
+++ b/ashmem/Makefile
@@ -14,5 +14,8 @@ all:
 install:
 	cp ashmem_linux.ko $(DESTDIR)/
 
+modules_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$$PWD modules_install
+
 clean:
 	rm -rf deps.h *.o *.ko *.mod.c *.symvers *.order .*.cmd .tmp_versions
diff --git a/binder/Makefile b/binder/Makefile
index 96de63a..c4fce2d 100644
--- a/binder/Makefile
+++ b/binder/Makefile
@@ -10,5 +10,8 @@ all:
 install:
 	cp binder_linux.ko $(DESTDIR)/
 
+modules_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$$PWD modules_install
+
 clean:
 	rm -rf deps.h *.o *.ko *.mod.c *.symvers *.order .*.cmd .tmp_versions
-- 
2.17.1

