From 731e025bdebc346f5361c1053bb89f2a24b5134a Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 5 May 2018 12:47:06 +0000
Subject: [PATCH] Fix dependencies for LuneOS

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 713df49..76dda56 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,8 +118,6 @@ IF(CMAKE_BUILD_TYPE MATCHES [cC][oO][vV][eE][rR][aA][gG][eE])
   SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -ftest-coverage -fprofile-arcs" )
 ENDIF(CMAKE_BUILD_TYPE MATCHES [cC][oO][vV][eE][rR][aA][gG][eE])
 
-find_package(GMock)
-
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fPIC")
 
 
@@ -138,7 +136,6 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/anbox/build/config.h.in
 
 add_subdirectory(external)
 add_subdirectory(src)
-add_subdirectory(tests)
 add_subdirectory(android)
 
 if (NOT "${HOST_CMAKE_C_COMPILER}" STREQUAL "")
