From 59d14cbfd9d3b04aa09db14e204f3d897767faa4 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Thu, 2 Jun 2022 10:06:15 +0200
Subject: [PATCH] Settings.cpp: Make org.webosports privileged as well

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/Settings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Settings.cpp b/src/Settings.cpp
index 8d938aa..05a0972 100644
--- a/src/Settings.cpp
+++ b/src/Settings.cpp
@@ -370,7 +370,7 @@ std::string Settings::getDefaultIcon(const std::string type)
 
 bool Settings::isPrivilegedSource(const std::string &callerId)
 {
-	if(callerId.find("com.palm.",0) == std::string::npos && callerId.find("com.webos.", 0) == std::string::npos && callerId.find("com.lge.",0) == std::string::npos)
+	if(callerId.find("com.palm.",0) == std::string::npos && callerId.find("com.webos.", 0) == std::string::npos && callerId.find("com.lge.",0) == std::string::npos && callerId.find("org.webosports.",0) == std::string::npos)
 	{
 		return false;
 	}
