From 60f07a3aec320b16a7b28b47445f79ad0f41ffd6 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Fri, 11 Oct 2013 11:11:12 +0200
Subject: [PATCH] maliitplatforminputcontext: use system instead of session bus

Upstream-Status: Inappropiate [configuration]

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 .../platforminputcontexts/maliit/qmaliitplatforminputcontext.cpp        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforminputcontexts/maliit/qmaliitplatforminputcontext.cpp b/src/plugins/platforminputcontexts/maliit/qmaliitplatforminputcontext.cpp
index 6d748d4..63c8298 100644
--- a/src/plugins/platforminputcontexts/maliit/qmaliitplatforminputcontext.cpp
+++ b/src/plugins/platforminputcontexts/maliit/qmaliitplatforminputcontext.cpp
@@ -136,7 +136,7 @@ enum MaliitEventRequestType {
 
 static QString maliitServerAddress()
 {
-    org::maliit::Server::Address serverAddress(QStringLiteral("org.maliit.server"), QStringLiteral("/org/maliit/server/address"), QDBusConnection::sessionBus());
+    org::maliit::Server::Address serverAddress(QStringLiteral("org.maliit.server"), QStringLiteral("/org/maliit/server/address"), QDBusConnection::systemBus());
 
     QString address(serverAddress.address());
 
-- 
1.8.1.2

