From 0cb5f510b7c5c5af4bb0d794e611b479d97d8dd9 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Mon, 22 Jan 2024 08:43:32 +0100
Subject: [PATCH] Fix incorrect include name

---
Upstream-Status: Inappropriate [LuneOS specific]

 src/player/UriPlayer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/player/UriPlayer.cpp b/src/player/UriPlayer.cpp
index 4c53429..66cb3b9 100644
--- a/src/player/UriPlayer.cpp
+++ b/src/player/UriPlayer.cpp
@@ -24,7 +24,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <gst/video/videooverlay.h>
-#include <gst/codecparsers/gsth264parse.h>
+#include <gst/codecparsers/gsth264parser.h>
 #include "ElementFactory.h"
 
 #include "parser/parser.h"
