From 8a831d860d49f7a96df253e5216797f57ae4346e Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 7 Apr 2022 17:30:29 +0000
Subject: [PATCH] AppLoader: add import path for QML apps

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Upstream-Status: Pending

---
 tools/runner/apploader.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/tools/runner/apploader.cpp b/tools/runner/apploader.cpp
index 03618e2..74c4e0f 100644
--- a/tools/runner/apploader.cpp
+++ b/tools/runner/apploader.cpp
@@ -26,6 +26,7 @@
 #include <QtCore/QJsonValue>
 #include "apploader.h"
 #include <QDebug>
+#include <QDir>
 
 #if defined(SMACK_ENABLED)
 #include <QCryptographicHash>
@@ -191,6 +192,20 @@ bool AppLoader::loadApplication(const QString &appId, const QString &mainQml, co
 
     m_engine.rootContext()->setContextProperty("params", params);
 
+    /* look for an appinfo.json in the parent folders of the qml file, and add that folder as a QML import path */
+    QDir qmlDir =  QFileInfo(QUrl(mainQml).toLocalFile()).absoluteDir();
+    QString applicationBasePath = qmlDir.absolutePath();
+    while(!qmlDir.isRoot()) {
+        if(qmlDir.exists("appinfo.json")) {
+            applicationBasePath = qmlDir.absolutePath();
+            break;
+        }
+        qmlDir = QDir(qmlDir.absolutePath() + "/..");
+        qmlDir.makeAbsolute();
+    }
+    m_engine.addImportPath(applicationBasePath);
+    qDebug() << "Added QML ImportPath " << applicationBasePath;
+
     // APP_ID envvar is used in multiple places, e.g WebOSQuickWindow and Service plugins.
     if (!appId.isEmpty()) {
         QCoreApplication::setApplicationName(appId);
