From f0dbb6e9f083c85d6bb7f24f6b3f4a3474878671 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 10 Apr 2022 11:08:44 +0000
Subject: [PATCH] Setup missing env variables

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending

---
 tools/runner/main.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/runner/main.cpp b/tools/runner/main.cpp
index 44009bf..bb2ae99 100644
--- a/tools/runner/main.cpp
+++ b/tools/runner/main.cpp
@@ -130,6 +130,11 @@ int main(int argc, char *argv[])
         qputenv("WEBOS_DEVICE_PIXEL_RATIO", QByteArray("auto"));
         qDebug("Set default WEBOS_DEVICE_PIXEL_RATIO as 'auto'");
     }
+    if (qEnvironmentVariableIsEmpty("QT_QPA_PLATFORM")) {
+        // Set "webos-wayland" by default
+        qputenv("QT_QPA_PLATFORM", QByteArray("webos-wayland"));
+        qDebug("Set default QT_QPA_PLATFORM as 'webos-wayland'");
+    }
 
     QGuiApplication app(argc, argv);
     QString mainQml("");
