From d0e2ce185746eb907bd067fa12d51f23e4c53045 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 25 Nov 2023 13:39:18 +0000
Subject: [PATCH] WebOSQuickWindow: use APP_ID env variable for appId

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 src/common/webosquickwindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/webosquickwindow.cpp b/src/common/webosquickwindow.cpp
index 3576d8f..ba8d29d 100644
--- a/src/common/webosquickwindow.cpp
+++ b/src/common/webosquickwindow.cpp
@@ -56,7 +56,7 @@ WebOSQuickWindow::WebOSQuickWindow(QWindow *parent)
     QObject::connect(this, &WebOSQuickWindow::visibleChanged,
                      this, &WebOSQuickWindow::updatePendingWindowProperties);
 
-    const QString &id = QCoreApplication::applicationName();
+    const QString id = qEnvironmentVariable("APP_ID", QCoreApplication::applicationName());
     m_windowProperties.insert(WP_APPID, id);
     setWindowProperty(WP_APPID, id);
 
-- 
2.34.1

