From d47bf2bed415d0bc0ff5d52582e1892067075515 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Sun, 30 Mar 2014 14:39:09 +0200
Subject: [PATCH 3/7] Avoid abort when poll descriptor is 0

* LP: #1092377

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/modules/alsa/alsa-mixer.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/modules/alsa/alsa-mixer.c b/src/modules/alsa/alsa-mixer.c
index 99de0ec..66188b8 100644
--- a/src/modules/alsa/alsa-mixer.c
+++ b/src/modules/alsa/alsa-mixer.c
@@ -165,6 +165,9 @@ static void defer_cb(pa_mainloop_api *a, pa_defer_event *e, void *userdata) {
     if (n < 0) {
         pa_log("snd_mixer_poll_descriptors_count() failed: %s", pa_alsa_strerror(n));
         return;
+    } else if (n == 0) {
+        pa_log("snd_mixer_poll_descriptors_count() equal 0");
+        return;
     }
     num_fds = (unsigned) n;
 
-- 
1.9.1

