From a6a3706cfbfca18ff348ed85fb13e79eb1b2a81a Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 1 Aug 2023 10:36:43 +0200
Subject: [PATCH] configure.ac: disable help2man

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Upstream-Status: Pending
---
 configure.ac | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index c4ee280..48747e8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -429,12 +429,7 @@ fi
 dnl ===================
 dnl Checks for help2man
 dnl ===================
-AC_CHECK_PROG([HAVE_HELP2MAN], [help2man], [yes], [no])
-if test "x$HAVE_HELP2MAN" = "xno"
-then
-    AC_MSG_NOTICE([help2man not found. Man documentation pages will not be built.])
-fi
-AM_CONDITIONAL([HAVE_HELP2MAN], [test "x$HAVE_HELP2MAN" = "xyes"])
+AM_CONDITIONAL([HAVE_HELP2MAN], [test "x$enable_documentation" = "xyes"])
 
 
 dnl ====================
