From 43812f7dc7802cab2e73ac9b4605c44eb5c26882 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 7 Apr 2022 17:30:29 +0000
Subject: [PATCH] AppLoader: add import path for QML apps

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Upstream-Status: Pending
---
 tools/runner/apploader.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/tools/runner/apploader.cpp b/tools/runner/apploader.cpp
index c5cfd14..bf67151 100644
--- a/tools/runner/apploader.cpp
+++ b/tools/runner/apploader.cpp
@@ -24,6 +24,7 @@
 #include <QtCore/QJsonDocument>
 #include <QtCore/QJsonObject>
 #include <QtCore/QJsonValue>
+#include <QDir>
 
 #include "apploader.h"
 
@@ -115,6 +116,20 @@ bool AppLoader::loadApplication(const QString &appId, const QString &mainQml, co
 
     m_engine.rootContext()->setContextProperty("params", params);
 
+    /* look for an appinfo.json in the parent folders of the qml file, and add that folder as a QML import path */
+    QDir qmlDir =  QFileInfo(QUrl(mainQml).toLocalFile()).absoluteDir();
+    QString applicationBasePath = qmlDir.absolutePath();
+    while(!qmlDir.isRoot()) {
+        if(qmlDir.exists("appinfo.json")) {
+            applicationBasePath = qmlDir.absolutePath();
+            break;
+        }
+        qmlDir = QDir(qmlDir.absolutePath() + "/..");
+        qmlDir.makeAbsolute();
+    }
+    m_engine.addImportPath(applicationBasePath);
+    qDebug() << "Added QML ImportPath " << applicationBasePath;
+
     // APP_ID envvar is used in multiple places, e.g WebOSQuickWindow and Service plugins.
     if (!appId.isEmpty()) {
         QCoreApplication::setApplicationName(appId);
