From 08af34d2ea918a98051e2d9ebe98883d7916e555 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 10 Apr 2022 11:08:44 +0000
Subject: [PATCH] Setup missing env variables

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 tools/runner/main.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/tools/runner/main.cpp b/tools/runner/main.cpp
index 17c5b0b..ea383dd 100644
--- a/tools/runner/main.cpp
+++ b/tools/runner/main.cpp
@@ -130,6 +130,21 @@ int main(int argc, char *argv[])
         qputenv("WEBOS_DEVICE_PIXEL_RATIO", QByteArray("auto"));
         qDebug("Set default WEBOS_DEVICE_PIXEL_RATIO as 'auto'");
     }
+    if (qEnvironmentVariableIsEmpty("XDG_RUNTIME_DIR")) {
+        // Set /tmp/luna-session" by default
+        qputenv("XDG_RUNTIME_DIR", QByteArray("/tmp/luna-session"));
+        qDebug("Set default XDG_RUNTIME_DIR as '/tmp/luna-session'");
+    }
+    if (qEnvironmentVariableIsEmpty("QT_QPA_PLATFORM")) {
+        // Set "webos-wayland" by default
+        qputenv("QT_QPA_PLATFORM", QByteArray("webos-wayland"));
+        qDebug("Set default QT_QPA_PLATFORM as 'webos-wayland'");
+    }
+    if (qEnvironmentVariableIsEmpty("QML_XHR_ALLOW_FILE_READ")) {
+        // Set "1" by default
+        qputenv("QML_XHR_ALLOW_FILE_READ", QByteArray("1"));
+        qDebug("Set default QML_XHR_ALLOW_FILE_READ as '1'");
+    }
 
     QGuiApplication app(argc, argv);
     QString mainQml("");
