From c825535fe2a8cd642abf4c03680be791ff7102ca Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 10 Apr 2022 11:08:44 +0000
Subject: [PATCH] Setup missing env variables

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

---
 tools/runner/main.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/tools/runner/main.cpp b/tools/runner/main.cpp
index 23a509a..896a040 100644
--- a/tools/runner/main.cpp
+++ b/tools/runner/main.cpp
@@ -130,6 +130,16 @@ int main(int argc, char *argv[])
         qputenv("WEBOS_DEVICE_PIXEL_RATIO", QByteArray("auto"));
         qDebug("Set default WEBOS_DEVICE_PIXEL_RATIO as 'auto'");
     }
+    if (qEnvironmentVariableIsEmpty("XDG_RUNTIME_DIR")) {
+        // Set "auto" by default
+        qputenv("XDG_RUNTIME_DIR", QByteArray("/tmp/luna-session"));
+        qDebug("Set default XDG_RUNTIME_DIR as '/tmp/luna-session'");
+    }
+    if (qEnvironmentVariableIsEmpty("QT_QPA_PLATFORM")) {
+        // Set "auto" by default
+        qputenv("QT_QPA_PLATFORM", QByteArray("webos-wayland"));
+        qDebug("Set default QT_QPA_PLATFORM as 'webos-wayland'");
+    }
 
     QGuiApplication app(argc, argv);
     QString mainQml("");
