From 26c9c4643dbd029cb8f5f1977b8b51412f8e302f Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 13 Oct 2022 20:33:02 +0000
Subject: [PATCH] AppLoader: use old syntax for quit connect

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 tools/runner/apploader.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/runner/apploader.cpp b/tools/runner/apploader.cpp
index bf67151..d6f1f1d 100644
--- a/tools/runner/apploader.cpp
+++ b/tools/runner/apploader.cpp
@@ -34,8 +34,8 @@ AppLoader::AppLoader (QObject * parent)
     : QObject (parent),
       m_component (new QQmlComponent(&m_engine, nullptr))
 {
-    QObject::connect(&m_engine, &QQmlEngine::quit,
-                     QCoreApplication::instance(), &QCoreApplication::quit);
+    QObject::connect(&m_engine, SIGNAL(quit()),
+                     QCoreApplication::instance(), SLOT(quit()));
 }
 
 AppLoader::~AppLoader()
-- 
2.34.1

