From f6064deaaf745fba92502e553c6071dbf3f82464 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 25 Mar 2023 14:47:49 +0000
Subject: [PATCH] WebOSQuickWindow: make setWindowPropery Q_INVOKABLE

Upstream-Status: Pending
---
 src/common/webosquickwindow.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/webosquickwindow.h b/src/common/webosquickwindow.h
index 3aa8ef3..231465c 100644
--- a/src/common/webosquickwindow.h
+++ b/src/common/webosquickwindow.h
@@ -158,6 +158,7 @@ public:
     bool cursorVisible() { return m_cursorVisible; }
 
     QObject *windowProperties() { return &m_windowProperties; }
+    Q_INVOKABLE void setWindowProperty(const QString& key, const QString& value);
 
     /*!
      * The parent class function is not virtual and we need to signal the
@@ -219,7 +220,6 @@ private:
     WebOSShellSurface* shellSurface();
 #endif
 
-    void setWindowProperty(const QString& key, const QString& value);
     QPoint m_mousePosition;
 
     bool m_keepAlive;
-- 
2.34.1

