diff --git a/src/rockhopper/touchpanel/emulator/touchpanel.c b/src/rockhopper/touchpanel/emulator/touchpanel.c
index 3460468..d31b5b5 100644
--- a/src/rockhopper/touchpanel/emulator/touchpanel.c
+++ b/src/rockhopper/touchpanel/emulator/touchpanel.c
@@ -169,8 +169,8 @@ static float scaleX, scaleY;
 /* Using hardcoded values for now ,since ioctl FBIOGET_VSCREENINFO 
    on /dev/fb0 is returning invalid values in qemux86 */
 
-#define SCREEN_HORIZONTAL_RES	1024
-#define SCREEN_VERTICAL_RES	768
+#define SCREEN_HORIZONTAL_RES	720
+#define SCREEN_VERTICAL_RES	1280
 
 static int
 init_touchpanel(void)
