From be41f7f74cbe4fa3aee539cdcdb5354e2e7a5e8c Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 3 Apr 2022 10:11:39 +0000
Subject: [PATCH 06/12] Add relaunch support for legacy enyo apps

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/web_page_base.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/core/web_page_base.cc b/src/core/web_page_base.cc
index 4286058..b47f72d 100644
--- a/src/core/web_page_base.cc
+++ b/src/core/web_page_base.cc
@@ -274,6 +274,12 @@ void WebPageBase::SendRelaunchEvent() {
       << "    var launchEvent=new CustomEvent('webOSRelaunch', { detail: "
       << detail << " });"
       << "    document.dispatchEvent(launchEvent);"
+      
+      << "    console.log('[WAM] call Mojo.relaunch() for legacy apps');"
+      << "    if(typeof Mojo !== 'undefined') {"
+      << "      PalmSystem.launchParams = JSON.stringify(" << detail << ");"
+      << "      Mojo.relaunch();"
+      << "    }"
       << "}, 1);";
   EvaluateJavaScript(relaunch_event.str().c_str());
 }
-- 
2.41.0.windows.3

