From fae0ef27d95b9dbe89af5f97611a684c6f051b62 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 3 Apr 2022 11:12:06 +0000
Subject: [PATCH] RunningApp: disable killer timeout for app relaunch

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

---
 src/base/RunningApp.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/base/RunningApp.cpp b/src/base/RunningApp.cpp
index 41a8d08..14e3734 100644
--- a/src/base/RunningApp.cpp
+++ b/src/base/RunningApp.cpp
@@ -241,8 +241,8 @@ void RunningApp::setLifeStatus(LifeStatus lifeStatus)
     // For example, youtube app can be foreground after 30 seconds at boottime
     // See more info here PLAT-101882.
     if (isTransition(m_lifeStatus)) {
-        if (m_lifeStatus == LifeStatus::LifeStatus_LAUNCHING) {
-            // Donot start killing timer in case of launching
+        if (m_lifeStatus == LifeStatus::LifeStatus_LAUNCHING || m_lifeStatus == LifeStatus::LifeStatus_RELAUNCHING) {
+            // Donot start killing timer in case of (re)launching
         } else if (m_lifeStatus == LifeStatus::LifeStatus_CLOSING) {
             // App should be closed within 1 second
             startKillingTimer(TIMEOUT_CLOSE);
