From c4e59d5fc96bf043deb60564079ae980d073ebe5 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Thu, 2 Jun 2022 10:03:59 +0200
Subject: [PATCH 08/10] AppDescription.h: Add org.webosports as privileged as
 well

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 src/base/AppDescription.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/base/AppDescription.h b/src/base/AppDescription.h
index 81d934b..62c9c3d 100644
--- a/src/base/AppDescription.h
+++ b/src/base/AppDescription.h
@@ -233,7 +233,8 @@ public:
     {
         if (m_appId.find("com.palm.") == 0 ||
             m_appId.find("com.webos.") == 0 ||
-            m_appId.find("com.lge.") == 0)
+            m_appId.find("com.lge.") == 0 ||
+            m_appId.find("org.webosports.") == 0)
             return true;
         return false;
     }
-- 
2.37.3.windows.1

