From ca59d2722cb956b568c4444593970f52938f54ae Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Tue, 21 Feb 2023 17:57:20 +0000
Subject: [PATCH] NativeContainer: configure native apps

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 src/bus/client/NativeContainer.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/bus/client/NativeContainer.cpp b/src/bus/client/NativeContainer.cpp
index b01a5ef..793f030 100644
--- a/src/bus/client/NativeContainer.cpp
+++ b/src/bus/client/NativeContainer.cpp
@@ -200,6 +200,14 @@ void NativeContainer::launch(RunningAppPtr runningApp, LunaTaskPtr lunaTask)
             runningApp->getLinuxProcess().addEnv("QT_QUICK_CONTROLS_STYLE", "QtQuick.Controls.LuneOS");
     }
 
+    if (AppType::AppType_Native == type) {
+        Logger::info(getClassName(), __FUNCTION__, runningApp->getAppId(), "Adding LuneOS environment");
+        if(runningApp->getLaunchPoint()->getAppDesc()->useLuneOSStyle())
+            runningApp->getLinuxProcess().addEnv("QT_QUICK_CONTROLS_STYLE", "QtQuick.Controls.LuneOS");
+        runningApp->getLinuxProcess().addEnv("QT_QPA_PLATFORM", "wayland-egl");
+        runningApp->getLinuxProcess().addEnv("XDG_RUNTIME_DIR", "/tmp/luna-session");
+    }
+
     runningApp->setLS2Name(Logger::format("%s-%d", runningApp->getAppId().c_str(), s_instanceCounter));
     if (RuntimeInfo::getInstance().getUser().empty())
         runningApp->getLinuxProcess().openStdFile(Logger::format("/var/log/%s-%d", runningApp->getAppId().c_str(), s_instanceCounter++));
-- 
2.34.1

