From 551b6aab7982e05f71c036c2a6c44edf1beb8c51 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Fri, 18 Aug 2023 15:52:01 +0000
Subject: [PATCH] Setup QT_WAYLAND_SHELL_INTEGRATION for webOs

Upstream-Status: Pending

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/bus/client/NativeContainer.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/bus/client/NativeContainer.cpp b/src/bus/client/NativeContainer.cpp
index 5419e94..cdfa404 100644
--- a/src/bus/client/NativeContainer.cpp
+++ b/src/bus/client/NativeContainer.cpp
@@ -194,6 +194,7 @@ void NativeContainer::launch(RunningAppPtr runningApp, LunaTaskPtr lunaTask)
     runningApp->getLinuxProcess().addEnv("LS2_NAME", Logger::format("%s-%d", runningApp->getAppId().c_str(), s_instanceCounter));
     // also force the use of webos waylandinputcontext plugin
     runningApp->getLinuxProcess().addEnv("QT_IM_MODULE", "wayland");
+    runningApp->getLinuxProcess().addEnv("QT_WAYLAND_SHELL_INTEGRATION", "webos");
 
     if (AppType::AppType_Native_Qml == type) {
         if(runningApp->getLaunchPoint()->getAppDesc()->useLuneOSStyle())
-- 
2.34.1

