From 4b68f670d4a837bdf3747ad5f2a3d9ebf1a2d8e6 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 17 Jul 2022 09:49:50 +0000
Subject: [PATCH] Setup QT_IM_MODULE for client apps

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

---
 src/bus/client/NativeContainer.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/bus/client/NativeContainer.cpp b/src/bus/client/NativeContainer.cpp
index 07af75a..b01a5ef 100644
--- a/src/bus/client/NativeContainer.cpp
+++ b/src/bus/client/NativeContainer.cpp
@@ -192,6 +192,8 @@ void NativeContainer::launch(RunningAppPtr runningApp, LunaTaskPtr lunaTask)
     runningApp->getLinuxProcess().addEnv("APP_ID", runningApp->getAppId());
     runningApp->getLinuxProcess().addEnv("DISPLAY_ID", std::to_string(runningApp->getDisplayId()));
     runningApp->getLinuxProcess().addEnv("LS2_NAME", Logger::format("%s-%d", runningApp->getAppId().c_str(), s_instanceCounter));
+    // also force the use of webos waylandinputcontext plugin
+    runningApp->getLinuxProcess().addEnv("QT_IM_MODULE", "wayland");
 
     if (AppType::AppType_Native_Qml == type) {
         if(runningApp->getLaunchPoint()->getAppDesc()->useLuneOSStyle())
