diff --git a/src/connman.service.in b/src/connman.service.in
index 48afb63..753e14c 100644
--- a/src/connman.service.in
+++ b/src/connman.service.in
@@ -1,13 +1,14 @@
 [Unit]
 Description=Connection service
-After=syslog.target
+After=syslog.target android-system.service
 Before=remote-fs.target
 
 [Service]
 Type=dbus
 BusName=net.connman
 Restart=on-failure
-ExecStart=@prefix@/sbin/connmand -n
+EnvironmentFile=/etc/connman/environment.conf
+ExecStart=@prefix@/sbin/connmand -n $EXTRA_PARAM
 StandardOutput=null
 
 [Install]
