diff -uNr connman-1.20.orig/src/connman.service.in connman-1.20/src/connman.service.in
--- connman-1.20.orig/src/connman.service.in	2013-02-26 13:31:20.000000000 +0100
+++ connman-1.20/src/connman.service.in	2013-12-31 19:04:32.156038358 +0100
@@ -6,7 +6,8 @@
 Type=dbus
 BusName=net.connman
 Restart=on-failure
-ExecStart=@prefix@/sbin/connmand -n
+EnvironmentFile=/etc/connman/environment.conf
+ExecStart=@prefix@/sbin/connmand -n $EXTRA_PARAM
 StandardOutput=null
 
 [Install]
