From 98c8823a2de1d41f59c3aad3c25996d09aebf842 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 4 Apr 2019 08:48:23 +0000
Subject: [PATCH] qtwebchannel mojom: provide a sync call for LuneOS

---
 chromium/qtwebengine/browser/qtwebchannel.mojom | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/chromium/qtwebengine/browser/qtwebchannel.mojom b/chromium/qtwebengine/browser/qtwebchannel.mojom
index 622e6a40421..f1fbf573a58 100644
--- a/chromium/qtwebengine/browser/qtwebchannel.mojom
+++ b/chromium/qtwebengine/browser/qtwebchannel.mojom
@@ -14,5 +14,8 @@ interface WebChannelTransportRender {
 
 interface WebChannelTransportHost {
   DispatchWebChannelMessage(array<uint8> binaryJson);
+  // Provide a sync call for LuneOS
+  [Sync]
+  DispatchWebChannelMessageSync(array<uint8> binaryJson) => (array<uint8> replyBinaryJson);
 };
 
