diff --git a/chromium/third_party/WebKit/Source/bindings/modules/v8/generated.gni b/chromium/third_party/WebKit/Source/bindings/modules/v8/generated.gni
index 0d8f27d..db9b850 100644
--- a/chromium/third_party/WebKit/Source/bindings/modules/v8/generated.gni
+++ b/chromium/third_party/WebKit/Source/bindings/modules/v8/generated.gni
@@ -46,6 +46,8 @@ bindings_modules_generated_union_type_files = [
   "$bindings_modules_v8_output_dir/RenderingContext.h",
   "$bindings_modules_v8_output_dir/RequestOrUSVString.cpp",
   "$bindings_modules_v8_output_dir/RequestOrUSVString.h",
+  "$bindings_modules_v8_output_dir/ServiceCallback.cpp",
+  "$bindings_modules_v8_output_dir/ServiceCallback.h",
   "$bindings_modules_v8_output_dir/ServiceWorkerOrMessagePort.cpp",
   "$bindings_modules_v8_output_dir/ServiceWorkerOrMessagePort.h",
   "$bindings_modules_v8_output_dir/StringOrArrayBufferOrNFCMessage.cpp",
diff --git a/chromium/third_party/WebKit/Source/modules/webos/BUILD.gn b/chromium/third_party/WebKit/Source/modules/webos/BUILD.gn
index effd211..2da4492 100644
--- a/chromium/third_party/WebKit/Source/modules/webos/BUILD.gn
+++ b/chromium/third_party/WebKit/Source/modules/webos/BUILD.gn
@@ -1,6 +1,11 @@
 # Copyright 2017 Herman van Hazendonk. All rights reserved.
 
 import("//third_party/WebKit/Source/modules/modules.gni")
+import("//build/config/linux/pkg_config.gni")
+
+pkg_config("ls2") {
+  packages = [ "luna-service2", "glib-2.0" ]
+}  
 
 blink_modules_sources("webos") {
   sources = [
@@ -10,4 +15,13 @@ blink_modules_sources("webos") {
     "PalmServiceBridge.cpp",
     "PalmServiceBridge.h",
   ]
+   
+  public_configs = [
+    ":ls2",
+  ]
+  
+  libs = [
+    "PmLogLib",   
+  ]
+
 }
diff --git a/chromium/third_party/WebKit/Source/modules/webos/LunaServiceMgr.cpp b/chromium/third_party/WebKit/Source/modules/webos/LunaServiceMgr.cpp
index 3562dc6..f4b9f60 100644
--- a/chromium/third_party/WebKit/Source/modules/webos/LunaServiceMgr.cpp
+++ b/chromium/third_party/WebKit/Source/modules/webos/LunaServiceMgr.cpp
@@ -47,7 +47,7 @@ void GMainContextIterate()
     base::MessageLoop *pCurrentMsgLoop = base::MessageLoop::current();
     if( doIterateNext && pCurrentMsgLoop )
     {
-        pCurrentMsgLoop->PostDelayedTask(FROM_HERE, base::Bind(&GMainContextIterate), base::TimeDelta::FromMilliseconds(100));
+        pCurrentMsgLoop->task_runner()->PostDelayedTask(FROM_HERE, base::Bind(&GMainContextIterate), base::TimeDelta::FromMilliseconds(100));
     }
 }
 
diff --git a/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.cpp b/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.cpp
index ac1775a..b6ece79 100644
--- a/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.cpp
+++ b/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.cpp
@@ -5,7 +5,6 @@
 #include "core/events/Event.h"
 #include "core/dom/ExceptionCode.h"
 #include "core/frame/Frame.h"
-#include "platform/Logging.h"
 #include "core/page/Page.h"
 #include "core/frame/Settings.h"
 #include <wtf/text/WTFString.h>
diff --git a/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.h b/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.h
index d508f87..46b6440 100644
--- a/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.h
+++ b/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.h
@@ -6,6 +6,7 @@
 #include "bindings/core/v8/ScriptWrappable.h"
 #include "bindings/core/v8/V8Binding.h"
 #include "bindings/core/v8/ScriptValue.h"
+#include "bindings/modules/v8/ServiceCallback.h"
 #include "core/events/Event.h"
 #include "core/events/EventListener.h"
 #include "core/events/EventTarget.h"
@@ -54,14 +55,11 @@ class PalmServiceBridge : public GarbageCollectedFinalized<PalmServiceBridge>,
         int call(const String& uri, const String& payload, ExceptionState&);
         void cancel();
 
-        void setOnservicecallback(ScriptValue& cbScriptValue) {
-            m_callbackScriptValue = cbScriptValue;
-            if (m_scriptState) {
-                m_scriptState->clear();
-            }
-            m_scriptState.reset(new ScriptStateProtectingContext(cbScriptValue.getScriptState()));
+        void setOnservicecallback(ScriptState *&cbState, ServiceCallback *cbValue) {
+            m_callbackScriptValue = ScriptValue::from(cbState, cbValue->v8Value(cbState->isolate()));
+            m_scriptState.reset(new ScriptStateProtectingContext(cbState));
         }
-        ScriptValue onservicecallback() const { return m_callbackScriptValue; }
+        ScriptWrappable *onservicecallback(ScriptState *cbState) const { return nullptr; }
 
         // callback from LunaServiceManagerListener
         virtual void serviceResponse(const char* body);
diff --git a/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.idl b/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.idl
index 2b1d28f..fc808d0 100644
--- a/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.idl
+++ b/chromium/third_party/WebKit/Source/modules/webos/PalmServiceBridge.idl
@@ -11,5 +11,5 @@ callback ServiceCallback = void (DOMString message);
     [RaisesException] unsigned long call(DOMString method, DOMString url);
     void cancel();
 
-    attribute ServiceCallback onservicecallback;
+    [CallWith=ScriptState] attribute ServiceCallback onservicecallback;
 };
