From e9fcd83639a9b7e2b1b0385b698e3901b58a1824 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 10 Aug 2019 15:13:49 +0000
Subject: [PATCH] VoiceCallManager: Fix DEVICELOCK handling

---
 src/voicecallmanager.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/voicecallmanager.cpp b/src/voicecallmanager.cpp
index 305d257..9e7a972 100644
--- a/src/voicecallmanager.cpp
+++ b/src/voicecallmanager.cpp
@@ -339,13 +339,11 @@ void VoiceCallManager::onVoiceCallAdded(AbstractVoiceCallHandler *handler)
 #ifdef WITH_NEMO_DEVICELOCK
     if (!handler->isEmergency()
             && d->deviceLock.state() == NemoDeviceLock::DeviceLock::ManagerLockout) {
-#else
-    if (!handler->isEmergency()) {
-#endif
+                
         handler->hangup();
         return;
-
     }
+#endif
 
     //AudioCallPolicyProxy *pHandler = new AudioCallPolicyProxy(handler, this);
     d->voiceCalls.insert(handler->handlerId(), handler);
-- 
2.17.0

