--- a/configure.ac.orig	2013-11-06 11:14:31.294097701 +0100
+++ b/configure.ac	2013-11-06 11:16:34.986363632 +0100
@@ -429,12 +429,7 @@
 dnl ===================
 dnl Checks for help2man
 dnl ===================
-AC_CHECK_PROG([HAVE_HELP2MAN], [help2man], [yes], [no])
-if test "x$HAVE_HELP2MAN" = "xno"
-then
-    AC_MSG_NOTICE([help2man not found. Man documentation pages will not be built.])
-fi
-AM_CONDITIONAL([HAVE_HELP2MAN], [test "x$HAVE_HELP2MAN" = "xyes"])
+AM_CONDITIONAL([HAVE_HELP2MAN], [test "x$enable_documentation" = "xyes"])
 
 
 dnl ====================
