From f0545b00454d0adbde8b0f1ef00fb1a773757f8b Mon Sep 17 00:00:00 2001
From: Herrie <merproject.org@herrie.org>
Date: Sat, 21 Oct 2017 05:02:38 +0000
Subject: [PATCH] Make DEPLOYMENT_PATH configurable. Other OS-es like LuneOS that also use Fingerterm might want to have it in another location like "/usr/palm/applications/$$TARGET" instead. Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

---
 fingerterm.pro | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/fingerterm.pro b/fingerterm.pro
index 0bb559e..adb8ff0 100644
--- a/fingerterm.pro
+++ b/fingerterm.pro
@@ -15,7 +15,10 @@ isEmpty(DEFAULT_FONT) {
     DEFAULT_FONT = monospace
 }
 
-DEPLOYMENT_PATH = /usr/share/$$TARGET
+isEmpty(DEPLOYMENT_PATH) {
+    DEPLOYMENT_PATH = /usr/share/$$TARGET
+}
+
 DEFINES += DEPLOYMENT_PATH=\\\"$$DEPLOYMENT_PATH\\\"
 DEFINES += DEFAULT_FONTFAMILY=\\\"$$DEFAULT_FONT\\\"
 
--
libgit2 0.24.0
