From a31e5e3692229ea9d10a3a8f24529fcc601551b2 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Fri, 21 Sep 2012 09:00:35 +0200
Subject: [PATCH] Disable use of luna-keymap-headers

---
 include/public/messages/SysMgrEvent.h |    5 -----
 1 file changed, 5 deletions(-)

diff --git a/include/public/messages/SysMgrEvent.h b/include/public/messages/SysMgrEvent.h
index e37a3b2..5ea27fb 100644
--- a/include/public/messages/SysMgrEvent.h
+++ b/include/public/messages/SysMgrEvent.h
@@ -87,12 +87,7 @@ struct SysMgrEvent
 		LastAttribute = 0xFFFF
 	};
 
-    // Look in the luna-keymap-headers package
-#if defined(TARGET_DESKTOP) || defined(TARGET_EMULATOR)
 	#include "SysMgrDeviceKeydefs.h"
-#else
-	#include <webosDeviceKeydefs.h>
-#endif
 
 	enum Orientation {
 		Orientation_Invalid = 0,
-- 
1.7.9.5

