diff --git a/Tools/qmake/mkspecs/features/features.prf b/Tools/qmake/mkspecs/features/features.prf
index 0f503df..a9ab3c6 100644
--- a/Tools/qmake/mkspecs/features/features.prf
+++ b/Tools/qmake/mkspecs/features/features.prf
@@ -11,6 +11,10 @@ defineTest(detectFeatures) {
 
     include(features.pri)
 
+    CONFIG += OE_QTLOCATION_ENABLED
+    CONFIG += OE_QTSENSORS_ENABLED
+    CONFIG += OE_GSTREAMER_ENABLED
+
     for(feature, FEATURE_DEFAULTS) {
         enabledFeature = $$find(feature, =1$)
         !isEmpty(enabledFeature) {
@@ -37,8 +41,8 @@ defineTest(detectFeatures) {
     qtHaveModule(printsupport): WEBKIT_CONFIG += have_qtprintsupport
     qtHaveModule(widgets): WEBKIT_CONFIG += have_qstyle
     qtHaveModule(testlib): WEBKIT_CONFIG += have_qttestlib
-    qtHaveModule(positioning): WEBKIT_CONFIG += have_qtpositioning
-    qtHaveModule(sensors): WEBKIT_CONFIG += have_qtsensors
+    OE_QTLOCATION_ENABLED:qtHaveModule(positioning): WEBKIT_CONFIG += have_qtpositioning
+    OE_QTSENSORS_ENABLED:qtHaveModule(sensors): WEBKIT_CONFIG += have_qtsensors
 
     config_libxml2: WEBKIT_CONFIG += use_libxml2
     config_libxslt: WEBKIT_CONFIG += xslt
@@ -99,15 +103,15 @@ defineTest(detectFeatures) {
 
     # HTML5 Media Support for builds with GStreamer
     unix:!mac:!contains(QT_CONFIG, no-pkg-config) {
-        packagesExist("glib-2.0 gio-2.0 gstreamer-1.0 gstreamer-plugins-base-1.0") {
+        OE_GSTREAMER_ENABLED:packagesExist("glib-2.0 gio-2.0 gstreamer-1.0 gstreamer-plugins-base-1.0") {
             WEBKIT_CONFIG += video use_gstreamer
-        } else: packagesExist("glib-2.0 gio-2.0 \'gstreamer-0.10 >= 0.10.30\' \'gstreamer-plugins-base-0.10 >= 0.10.30\'") {
+        } else: OE_GSTREAMER010_ENABLED:packagesExist("glib-2.0 gio-2.0 \'gstreamer-0.10 >= 0.10.30\' \'gstreamer-plugins-base-0.10 >= 0.10.30\'") {
             WEBKIT_CONFIG += video use_gstreamer use_gstreamer010
         }
         use?(gstreamer): WEBKIT_CONFIG += use_native_fullscreen_video
     }
 
-    !enable?(video):qtHaveModule(multimediawidgets) {
+    !enable?(video):OE_QTMULTIMEDIA_ENABLED:qtHaveModule(multimediawidgets) {
         WEBKIT_CONFIG += video use_qt_multimedia
     }
 
