From 043d3d0d41e90ca73e17d6691e6c0de90bf5b1c3 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 17 Dec 2018 09:02:57 +0000
Subject: [PATCH] tests: add vboxvideo to tested kernel modules

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 tests/util/kms.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/util/kms.c b/tests/util/kms.c
index 08b48fe5..7cf9d9d6 100644
--- a/tests/util/kms.c
+++ b/tests/util/kms.c
@@ -149,6 +149,7 @@ static const char * const modules[] = {
 	"armada-drm",
 	"komeda",
 	"imx-dcss",
+	"vboxvideo",
 };
 
 int util_open(const char *device, const char *module)
