From 2ef7842cfd7c30b6f446fe7578e4d61973359f96 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 17 Dec 2018 09:02:57 +0000
Subject: [PATCH] tests: add vboxvideo to tested kernel modules

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 tests/util/kms.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/util/kms.c b/tests/util/kms.c
index dd1bbee3..cc6a7e70 100644
--- a/tests/util/kms.c
+++ b/tests/util/kms.c
@@ -147,6 +147,7 @@ static const char * const modules[] = {
 	"stm",
 	"sun4i-drm",
 	"armada-drm",
+	"vboxvideo",
 };
 
 int util_open(const char *device, const char *module)
