From 230e7097875b0d1f5203a5eaff2e382e752010ad Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 17 Dec 2018 09:02:57 +0000
Subject: [PATCH] tests: add vboxvideo to tested kernel modules

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 tests/util/kms.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/util/kms.c b/tests/util/kms.c
index 5af8bf20..5d015181 100644
--- a/tests/util/kms.c
+++ b/tests/util/kms.c
@@ -146,6 +146,7 @@ static const char * const modules[] = {
 	"pl111",
 	"stm",
 	"sun4i-drm",
+	"vboxvideo",
 };
 
 int util_open(const char *device, const char *module)
-- 
2.17.1

