From 10a36adcfbecea40545eeb54d57b7b42debd9351 Mon Sep 17 00:00:00 2001
From: Herrie <github.com@herrie.org>
Date: Wed, 9 Mar 2016 18:20:28 +0100
Subject: [PATCH] Add dbus policy for Bluez4

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 src/daemon/pulseaudio-system.conf | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/src/daemon/pulseaudio-system.conf b/src/daemon/pulseaudio-system.conf
index 2eb342e..ac9b60e 100644
--- a/src/daemon/pulseaudio-system.conf
+++ b/src/daemon/pulseaudio-system.conf
@@ -24,8 +24,36 @@ License along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
   <!-- System-wide PulseAudio runs as 'pulse' user. This fragment is
        not necessary for user PulseAudio instances. -->
 
+  <policy user="root">
+    <allow own="org.pulseaudio.Server"/>
+    <allow send_destination="org.bluez"/>
+    <allow send_interface="org.bluez.Manager"/>
+    <allow send_interface="org.bluez.Audio"/>
+    <allow send_interface="org.bluez.AudioSink"/>
+    <allow send_interface="org.bluez.AudioSource"/>
+    <allow send_interface="org.bluez.Headset"/>
+    <allow send_interface="org.bluez.HeadsetGateway"/>
+    <allow send_interface="org.bluez.MediaEndpoint"/>
+    <allow send_interface="org.freedesktop.DBus.ObjectManager"/>  
+  </policy>
+  
   <policy user="pulse">
     <allow own="org.pulseaudio.Server"/>
+    <allow send_destination="org.bluez"/>
+    <allow send_interface="org.bluez.Manager"/>
+    <allow send_interface="org.bluez.Audio"/>
+    <allow send_interface="org.bluez.AudioSink"/>
+    <allow send_interface="org.bluez.AudioSource"/>
+    <allow send_interface="org.bluez.Headset"/>
+    <allow send_interface="org.bluez.HeadsetGateway"/>
+    <allow send_interface="org.bluez.MediaEndpoint"/>
+    <allow send_interface="org.freedesktop.DBus.ObjectManager"/>  
+  </policy>
+  
+  <policy context="default">
+    <deny own="org.pulseaudio.Server"/>
+    <deny send_destination="org.bluez"/>
+    <deny send_interface="org.bluez.Manager"/>
   </policy>
 
 </busconfig>
