=== modified file 'server/UbuntuMtpDatabase.h'
--- server/UbuntuMtpDatabase.h	2013-09-13 17:11:57 +0000
+++ server/UbuntuMtpDatabase.h	2013-11-13 14:57:19 +0000
@@ -133,12 +133,8 @@
 
 	db = std::map<MtpObjectHandle, DbEntry>();
 
-	readFiles(basedir + "/Documents");
-	readFiles(basedir + "/Music");
-	readFiles(basedir + "/Videos");
-	readFiles(basedir + "/Pictures");
-	readFiles(basedir + "/Downloads");
-	
+	readFiles(basedir + "/");
+
         std::cout << __PRETTY_FUNCTION__ << ": object count:" << db.size() << std::endl;
     }
 

=== modified file 'server/server.cpp'
--- server/server.cpp	2013-09-13 16:37:39 +0000
+++ server/server.cpp	2013-11-13 14:57:42 +0000
@@ -58,7 +58,7 @@
     {
         new android::MtpServer(
             fd, 
-            new android::UbuntuMtpDatabase(userdata->pw_dir),
+            new android::UbuntuMtpDatabase("/media/internal"),
             false, 
             userdata->pw_gid, 
             FileSystemConfig::file_perm, 

