From 29ec695807693371dc18409b1ec1ea1d17b85163 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Fri, 31 Jan 2014 16:28:02 +0100
Subject: [PATCH] Don't use internal wayland based input context

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/client/qwaylandintegration.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/client/qwaylandintegration.cpp b/src/client/qwaylandintegration.cpp
index b354ac0..37b7443 100644
--- a/src/client/qwaylandintegration.cpp
+++ b/src/client/qwaylandintegration.cpp
@@ -122,7 +122,10 @@ QWaylandIntegration::QWaylandIntegration()
     foreach (QPlatformScreen *screen, mDisplay->screens())
         screenAdded(screen);
 
-    mInputContext.reset(new QWaylandInputContext(mDisplay));
+    if (!qgetenv("QT_WAYLAND_INPUT_CONTEXT").isEmpty())
+        mInputContext.reset(new QWaylandInputContext(mDisplay));
+    else
+        mInputContext.reset(QPlatformInputContextFactory::create());
 }
 
 QWaylandIntegration::~QWaylandIntegration()
-- 
1.8.3.2

