From 600c2a0b5e95076d2d64c89609ca3f3192a0dc54 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Wed, 23 Feb 2022 18:31:46 +0000
Subject: [PATCH] Setup service for LuneOS

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Upstream-Status: Pending
---
 unit/vmwgfx-setlayout.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/unit/vmwgfx-setlayout.cmake b/unit/vmwgfx-setlayout.cmake
index b11fe6a..b3e36ae 100644
--- a/unit/vmwgfx-setlayout.cmake
+++ b/unit/vmwgfx-setlayout.cmake
@@ -17,11 +17,13 @@
 [Unit]
 Description=webos - "%n"
 DefaultDependencies=no
+Before=surface-manager.service
 
 [Install]
-WantedBy=default.target
+WantedBy=multi-user.target
 
 [Service]
 Type=oneshot
+ExecStartPre=/sbin/modprobe vmwgfx
 ExecStart=${CMAKE_INSTALL_PREFIX}/bin/vmwgfx-setlayout
 RemainAfterExit=no
