From 90d6c2bce5a50d2c05af25978bccbe047705eed7 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Mon, 29 Dec 2014 12:05:20 +0100
Subject: [PATCH] Search within OE_QMAKE_PATH_EXTERNAL_HOST_BINS for
 qtwaylandscanner

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 find-modules/FindQtWaylandScanner.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/find-modules/FindQtWaylandScanner.cmake b/find-modules/FindQtWaylandScanner.cmake
index 7fb8a00..f782275 100644
--- a/find-modules/FindQtWaylandScanner.cmake
+++ b/find-modules/FindQtWaylandScanner.cmake
@@ -88,7 +88,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/ECMFindModuleHelpersStub.cmake)
 ecm_find_package_version_check(QtWaylandScanner)
 
 # Find qtwaylandscanner
-find_program(QtWaylandScanner_EXECUTABLE NAMES qtwaylandscanner)
+find_program(QtWaylandScanner_EXECUTABLE NAMES qtwaylandscanner HINTS ${OE_QMAKE_PATH_EXTERNAL_HOST_BINS})
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(QtWaylandScanner
-- 
2.1.0

