From 7983ac5566497e0e0c0d1aa1f0d01d2939c5f839 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Thu, 2 Jun 2022 10:07:52 +0200
Subject: [PATCH] DownloadManager.cpp: Make org.webosports privileged as
 well

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Pending
---
 src/DownloadManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/DownloadManager.cpp b/src/DownloadManager.cpp
index 4c227de..527b14c 100644
--- a/src/DownloadManager.cpp
+++ b/src/DownloadManager.cpp
@@ -220,7 +220,7 @@ bool DownloadManager::isPathInMedia(const std::string& path) {
 
 bool DownloadManager::isPrivileged(const std::string& sender)
 {
-    if(sender.find("com.palm.") == 0 || sender.find("com.webos.") == 0 || sender.find("com.lge.") == 0 )
+    if(sender.find("com.palm.") == 0 || sender.find("com.webos.") == 0 || sender.find("com.lge.") == 0 || sender.find("org.webosports.") == 0 )
         return true;
 
     return false;

