From 2dd693b083109bb4a34887806d04881ef7630b79 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Tue, 2 Feb 2016 18:21:58 +0100
Subject: [PATCH] external_string.h: make it compatible with node 4.2 and newer

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 src/external_string.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/external_string.h b/src/external_string.h
index 08bfe90..ecfe223 100644
--- a/src/external_string.h
+++ b/src/external_string.h
@@ -28,7 +28,7 @@
 
 v8::Local<v8::String> createV8StringFromFile(const char* pathToFile);
 
-class MappedRegionExternalString : public v8::String::ExternalAsciiStringResource {
+class MappedRegionExternalString : public v8::String::ExternalOneByteStringResource {
 public:
 	static v8::Local<v8::String> create(const char* pathToFile);
 	MappedRegionExternalString(const char* pathToFile);
-- 
2.7.0

