diff --git a/maliit-keyboard/plugin/inputmethod.cpp b/maliit-keyboard/plugin/inputmethod.cpp
index 9a0e187..10b4d91 100644
--- a/maliit-keyboard/plugin/inputmethod.cpp
+++ b/maliit-keyboard/plugin/inputmethod.cpp
@@ -100,7 +102,7 @@ QQuickView *getOverlaySurface (MAbstractInputMethodHost *host, QQuickView *paren
 {
     QScopedPointer<QQuickView> view(new QQuickView (0));

-    view->setTransientParent(parent);
+    // view->setTransientParent(parent);

     host->registerWindow (view.data(), Maliit::PositionOverlay);

@@ -377,8 +379,8 @@ void InputMethod::show()
                                         d->layout.model.height())));

     d->surface->show();
-    d->extended_surface->show();
-    d->magnifier_surface->show();
+    //d->extended_surface->show();
+    //d->magnifier_surface->show();
 }

 void InputMethod::hide()
@@ -387,8 +389,8 @@ void InputMethod::hide()
     d->layout.updater.resetOnKeyboardClosed();
     d->editor.clearPreedit();
     d->surface->hide();
-    d->extended_surface->hide();
-    d->magnifier_surface->hide();
+    //d->extended_surface->hide();
+    //d->magnifier_surface->hide();
 }

 void InputMethod::setPreedit(const QString &preedit,
