From 8e99fe6caace3523b4bad3430b54b143ebdae25c Mon Sep 17 00:00:00 2001
From: Herrie <github.com@herrie.org>
Date: Sat, 2 Jul 2016 07:36:15 +0200
Subject: [PATCH] Extend ICQ password length

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 libpurple/protocols/oscar/oscar.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libpurple/protocols/oscar/oscar.h b/libpurple/protocols/oscar/oscar.h
index 9348071..e29d2b9 100644
--- a/libpurple/protocols/oscar/oscar.h
+++ b/libpurple/protocols/oscar/oscar.h
@@ -110,7 +110,7 @@ extern "C" {
 /**
  * Maximum length for the password of an ICQ account
  */
-#define MAXICQPASSLEN 8
+#define MAXICQPASSLEN 16
 
 #define AIM_MD5_STRING "AOL Instant Messenger (SM)"
 

