From de5576b3160516c7702079c3383984e0bae41429 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Fri, 12 Dec 2014 14:25:19 +0100
Subject: [PATCH] gdbus-codegen: replace plus also with underscore

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 gio/gdbus-2.0/codegen/codegen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/gdbus-2.0/codegen/codegen.py b/gio/gdbus-2.0/codegen/codegen.py
index 43a6837..6ffbcdf 100644
--- a/gio/gdbus-2.0/codegen/codegen.py
+++ b/gio/gdbus-2.0/codegen/codegen.py
@@ -47,7 +47,7 @@ class CodeGenerator:
             self.ns_upper = ''
             self.ns_lower = ''
         self.interface_prefix = interface_prefix
-        self.header_guard = self.h.name.upper().replace('.', '_').replace('-', '_').replace('/', '_')
+        self.header_guard = self.h.name.upper().replace('.', '_').replace('-', '_').replace('/', '_').replace('+', '_')
 
     # ----------------------------------------------------------------------------------------------------
 
-- 
2.1.0

