From de5576b3160516c7702079c3383984e0bae41429 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Fri, 12 Dec 2014 14:25:19 +0100
Subject: [PATCH] gdbus-codegen: replace plus also with underscore

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 gio/gdbus-2.0/codegen/codegen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff -uNr glib-2.52.2.orig/gio/gdbus-2.0/codegen/codegen.py glib-2.52.2/gio/gdbus-2.0/codegen/codegen.py
--- glib-2.52.2.orig/gio/gdbus-2.0/codegen/codegen.py	2017-07-21 19:01:58.055221558 +0200
+++ glib-2.52.2/gio/gdbus-2.0/codegen/codegen.py	2017-07-21 19:03:26.343220373 +0200
@@ -51,6 +51,8 @@
             self.ns_lower = ''
         self.interface_prefix = interface_prefix
         self.header_guard = header_name.upper().replace('.', '_').replace('-', '_').replace('/', '_')
+        self.header_guard = self.h.name.upper().replace('.', '_').replace('-', '_').replace('/', '_').replace('+', '_')
+
 
     # ----------------------------------------------------------------------------------------------------
 
