diff --git a/src/plugins/platforms/palm/palm.pro b/src/plugins/platforms/palm/palm.pro
index 840d22d..1d3c6e5 100644
--- a/src/plugins/platforms/palm/palm.pro
+++ b/src/plugins/platforms/palm/palm.pro
@@ -18,7 +18,7 @@ HEADERS += hidd_qpa.h \
            ScreenEdgeFlickGesture.h
 
 webos {
-    qemu* {
+    qemu*|tuna {
         include(../fb_base/fb_base.pri)
         SOURCES += ../linuxfb/qlinuxfbintegration.cpp \
                    emulatorfbintegration.cpp
