diff --git a/src/plugins/platforms/platforms.pro b/src/plugins/platforms/platforms.pro
index 7d7c103..1b38290 100644
--- a/src/plugins/platforms/platforms.pro
+++ b/src/plugins/platforms/platforms.pro
@@ -8,12 +8,6 @@ webos {
     desktop {
         SUBDIRS += xcb
         DEFINES += TARGET_DESKTOP
-    } else:qemu* {
-        SUBDIRS += palm
-    } else {
-        SUBDIRS += palm
-        SUBDIRS += webos
-        DEFINES += TARGET_DEVICE
     }
 }
 
