diff --git a/src/plugins/platforms/palm/palm.pro b/src/plugins/platforms/palm/palm.pro
index 840d22d..c8f4e83 100644
--- a/src/plugins/platforms/palm/palm.pro
+++ b/src/plugins/platforms/palm/palm.pro
@@ -42,7 +42,7 @@ webos {
                    qeglfswindowsurface.h \
                    qeglfsscreen.h
         DEFINES += TARGET_DEVICE
-        LIBS_PRIVATE += -lnyx -lhid -ldl
+        LIBS_PRIVATE += -lnyx -ldl
     }
 }
 
