diff --git a/src/plugins/platforms/palm/hiddtp_qpa.cpp b/src/plugins/platforms/palm/hiddtp_qpa.cpp
index 0019866..8f5bd77 100644
--- a/src/plugins/platforms/palm/hiddtp_qpa.cpp
+++ b/src/plugins/platforms/palm/hiddtp_qpa.cpp
@@ -21,6 +21,7 @@
 ****************************************************************************/
 
 
+#include <unistd.h>
 #include <fcntl.h>
 #include <sys/socket.h>
 #include <sys/un.h>
diff --git a/src/plugins/platforms/palm/nyxkeyboardhandler.cpp b/src/plugins/platforms/palm/nyxkeyboardhandler.cpp
index 766763c..4517fff 100644
--- a/src/plugins/platforms/palm/nyxkeyboardhandler.cpp
+++ b/src/plugins/platforms/palm/nyxkeyboardhandler.cpp
@@ -19,11 +19,11 @@
 #include <QKbdDriverFactory>
 #include <QDebug>
 #include <Qt>
+#include <unistd.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <errno.h>
-#include <../kernel/include/linux/compiler.h>
-#include <../kernel/include/linux/input.h> //to include the right input file
+#include <linux/input.h> //to include the right input file
 
 #include "nyxkeyboardhandler.h"
 #include "NyxInputControl.h"
