diff --git a/src/plugins/platforms/palm/palm.pro b/src/plugins/platforms/palm/palm.pro
index 66f357c..4808099 100644
--- a/src/plugins/platforms/palm/palm.pro
+++ b/src/plugins/platforms/palm/palm.pro
@@ -17,35 +17,13 @@ HEADERS += hidd_qpa.h \
            FlickGesture.h \
            ScreenEdgeFlickGesture.h
 
-webos {
-    qemux86 {
-        include(../fb_base/fb_base.pri)
-        SOURCES += ../linuxfb/qlinuxfbintegration.cpp \
-                   emulatorfbintegration.cpp
-        HEADERS += ../linuxfb/qlinuxfbintegration.h \
-                   emulatorfbintegration.h
-
-        LIBS_PRIVATE += -lnyx
-    } else {
-        QT += opengl
-        SOURCES += qeglfsintegration.cpp \
-                   ../eglconvenience/qeglconvenience.cpp \
-                   ../eglconvenience/qeglplatformcontext.cpp \
-                   qeglfswindow.cpp \
-                   qeglfswindowsurface.cpp \
-                   qeglfsscreen.cpp
-
-        HEADERS += qeglfsintegration.h \
-                   ../eglconvenience/qeglconvenience.h \
-                   ../eglconvenience/qeglplatformcontext.h \
-                   qeglfswindow.h \
-                   qeglfswindowsurface.h \
-                   qeglfsscreen.h
-        DEFINES += TARGET_DEVICE
-        LIBS_PRIVATE += -lnyx -lhid -ldl
-    }
-}
+include(../fb_base/fb_base.pri)
+SOURCES += ../linuxfb/qlinuxfbintegration.cpp \
+           emulatorfbintegration.cpp
+HEADERS += ../linuxfb/qlinuxfbintegration.h \
+           emulatorfbintegration.h
 
+LIBS_PRIVATE += -lnyx
 
 INCLUDEPATH += ../clipboards
 SOURCES += ../clipboards/qwebosclipboard.cpp
