diff --git a/src/gui/embedded/qkbd_qws.h b/src/gui/embedded/qkbd_qws.h
index 6eb815c..302efc9 100644
--- a/src/gui/embedded/qkbd_qws.h
+++ b/src/gui/embedded/qkbd_qws.h
@@ -53,6 +53,7 @@ QT_MODULE(Gui)
 
 #ifndef QT_NO_QWS_KEYBOARD
 
+
 class QWSKbPrivate;
 
 class  Q_GUI_EXPORT QWSKeyboardHandler
@@ -65,6 +66,7 @@ public:
     virtual void processKeyEvent(int unicode, int keycode, Qt::KeyboardModifiers modifiers,
                             bool isPress, bool autoRepeat);
 
+#ifndef QKBD_QWS_DIRTY_HACK_JUST_FOR_COMPILATION
     enum KeycodeAction {
         None               = 0,
 
@@ -85,6 +87,10 @@ public:
     };
 
     KeycodeAction processKeycode(quint16 keycode, bool pressed, bool autorepeat);
+#else
+    int processKeycode(quint16 keycode, bool pressed, bool autorepeat);
+#endif
+
 #if defined(QT_WEBOS)
     void setIsExternalKeyboard(bool val);
 #endif
diff --git a/src/plugins/platforms/palm/nyxkeyboardhandler.h b/src/plugins/platforms/palm/nyxkeyboardhandler.h
index ac845f4..39da1cd 100644
--- a/src/plugins/platforms/palm/nyxkeyboardhandler.h
+++ b/src/plugins/platforms/palm/nyxkeyboardhandler.h
@@ -18,6 +18,7 @@
 #ifndef __NYXKEYBOARDHANDLER_H__
 #define __NYXKEYBOARDHANDLER_H__
 
+#define QKBD_QWS_DIRTY_HACK_JUST_FOR_COMPILATION
 #include <QWSKeyboardHandler>
 #include <QWSMouseHandler>
 #include <qsocketnotifier.h>
