diff --git a/src/plugins/platforms/palm/qeglfsintegration.cpp b/src/plugins/platforms/palm/qeglfsintegration.cpp
index 49ba634..a40e76e 100644
--- a/src/plugins/platforms/palm/qeglfsintegration.cpp
+++ b/src/plugins/platforms/palm/qeglfsintegration.cpp
@@ -68,7 +68,6 @@ QEglFSIntegration::QEglFSIntegration(bool soft)
 {
     m_primaryScreen = new QEglFSScreen(EGL_DEFAULT_DISPLAY);
     m_tpHandler = new QPAHiddTpHandler(m_primaryScreen->geometry().width(), m_primaryScreen->geometry().height());
-    m_keyboard = new NyxKeyboardHandler;
     this->soft = soft;
     mScreens.append(m_primaryScreen);
 #ifdef QEGL_EXTRA_DEBUG
diff --git a/src/plugins/platforms/palm/qeglfsintegration.h b/src/plugins/platforms/palm/qeglfsintegration.h
index 69d5fb7..f46da5d 100644
--- a/src/plugins/platforms/palm/qeglfsintegration.h
+++ b/src/plugins/platforms/palm/qeglfsintegration.h
@@ -46,7 +46,6 @@
 #include "qeglfsscreen.h"
 
 #include "hiddtp_qpa.h"
-#include "nyxkeyboardhandler.h"
 
 #include "qwebosclipboard.h"
 #include <QtGui/QPlatformIntegration>
@@ -76,7 +75,6 @@ private:
     QEglFSScreen *m_primaryScreen;
 
     QPAHiddTpHandler* m_tpHandler;
-    NyxKeyboardHandler* m_keyboard;
 
     bool soft;
     QWebOSClipboard* m_clipboard;
