From 9c66dfac85aa067edf5dff76db7aa0c7b79712c1 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 4 May 2017 16:44:51 +0000
Subject: [PATCH] allow to build on aarch64

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/jemalloc_linux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jemalloc_linux.c b/src/jemalloc_linux.c
index 400d8ab..7a9dcbc 100644
--- a/src/jemalloc_linux.c
+++ b/src/jemalloc_linux.c
@@ -245,7 +245,7 @@
 #  define SIZEOF_PTR_2POW	3
 #  define CPU_SPINWAIT		__asm__ volatile("pause")
 #endif
-#ifdef __arm__
+#if defined(__arm__) || defined (__aarch64__)
 #  define PAGESIZE_2POW		12
 #  define QUANTUM_2POW		3
 #  define SIZEOF_PTR_2POW	2
-- 
1.9.1

