From 27d53312422e618166dd7dc0b3db95d62fb8c4df Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 23 Apr 2022 14:49:30 +0000
Subject: [PATCH] Correctly detect wayland platform

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 passthroughserver/main.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/passthroughserver/main.cpp b/passthroughserver/main.cpp
index f8d415b..364a0fc 100644
--- a/passthroughserver/main.cpp
+++ b/passthroughserver/main.cpp
@@ -134,7 +134,7 @@ QSharedPointer<MInputContextConnection> createConnection(const MImServerConnecti
 {
     Q_UNUSED(options);
 #ifdef HAVE_WAYLAND
-    if (QGuiApplication::platformName().startsWith("wayland")) {
+    if (QGuiApplication::platformName().contains("wayland")) {
         return QSharedPointer<MInputContextConnection>(Maliit::createWestonIMProtocolConnection());
     }
 #endif
@@ -146,7 +146,7 @@ QSharedPointer<MInputContextConnection> createConnection(const MImServerConnecti
 QSharedPointer<Maliit::AbstractPlatform> createPlatform()
 {
 #ifdef HAVE_WAYLAND
-    if (QGuiApplication::platformName().startsWith("wayland")) {
+    if (QGuiApplication::platformName().contains("wayland")) {
         return QSharedPointer<Maliit::AbstractPlatform>(new Maliit::WaylandPlatform);
     } else
 #endif
-- 
2.17.1

