From 24b00c5cee9fe47c749e63d282dda02383355205 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Fri, 22 Jul 2022 12:28:03 +0200
Subject: [PATCH] 31-webos-aliases.conf: Use Prelude for LuneOS

Upstream-Status: Pending
---
 31-webos-aliases.conf | 43 ++++++++++++++++++++++---------------------
 1 file changed, 22 insertions(+), 21 deletions(-)

diff --git a/31-webos-aliases.conf b/31-webos-aliases.conf
index e428a85..5bef18d 100644
--- a/31-webos-aliases.conf
+++ b/31-webos-aliases.conf
@@ -5,8 +5,7 @@
     <!-- webOS default fallback font -->
     <match>
         <edit mode="append" name="family">
-            <string>Museo Sans</string>
-            <string>Noto Sans</string>
+            <string>Prelude</string>
         </edit>
     </match>
 
@@ -15,7 +14,7 @@
             <string>Museo SansLight</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>thin</const>
@@ -26,7 +25,7 @@
             <string>Museo SansNormal</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>light</const>
@@ -37,7 +36,7 @@
             <string>Museo SansDemiBold</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>medium</const>
@@ -48,7 +47,7 @@
             <string>Museo SansBold</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>bold</const>
@@ -59,7 +58,7 @@
             <string>Museo SansBlack</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>black</const>
@@ -70,7 +69,7 @@
             <string>Museo SansDemiBoldItalic</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>medium</const>
@@ -84,7 +83,7 @@
             <string>Museo SansBoldItalic</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>bold</const>
@@ -98,7 +97,7 @@
             <string>Museo SansBlackItalic</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>black</const>
@@ -108,6 +107,7 @@
         </edit>
     </match>
 
+<!-- FIXME: Disable language specific fonts for now for LuneOS
     <match target="pattern">
         <test name="lang" compare="contains" qual="any">
             <string>ar</string>
@@ -396,6 +396,7 @@
             <string>Noto Sans CJK KR</string>
         </edit>
     </match>
+-->
 
     <!--
     Edit the fonts that are matched by the patterns above
@@ -481,7 +482,7 @@
             <string>Moon Miso Bold</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Miso</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>bold</const>
@@ -492,7 +493,7 @@
             <string>Moon Miso Light</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Miso</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>light</const>
@@ -503,7 +504,7 @@
             <string>Moon Miso Medium</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Miso</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>regular</const>
@@ -514,7 +515,7 @@
             <string>Moon Museo Sans Thin</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>thin</const>
@@ -525,7 +526,7 @@
             <string>Moon Museo Sans Light</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>light</const>
@@ -536,7 +537,7 @@
             <string>Moon Museo Sans Medium</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>medium</const>
@@ -547,7 +548,7 @@
             <string>Moon Museo Sans Medium Italic</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>medium</const>
@@ -561,7 +562,7 @@
             <string>Moon Museo Sans Bold</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>bold</const>
@@ -572,7 +573,7 @@
             <string>Moon Museo Sans Bold Italic</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>bold</const>
@@ -586,7 +587,7 @@
             <string>Moon Museo Sans Black</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>black</const>
@@ -597,7 +598,7 @@
             <string>Moon Museo Sans Black Italic</string>
         </test>
         <edit name="family" mode="assign" binding="same">
-            <string>Museo Sans</string>
+            <string>Prelude</string>
         </edit>
         <edit name="weight" mode="assign" binding="same">
             <const>black</const>
-- 
2.31.0.windows.1

