From 04e5f275a01a0a62981d2e0b3a406cbd1e86bf2e Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 10 Sep 2013 14:41:44 +0200
Subject: [PATCH 4/4] Change path to QtWebProcess executable in pub/prv hub
 configuration

Upstream-Status: Inappropiate [configuration]

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 files/conf/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/files/conf/CMakeLists.txt b/files/conf/CMakeLists.txt
index 16b7ead..0666d04 100644
--- a/files/conf/CMakeLists.txt
+++ b/files/conf/CMakeLists.txt
@@ -43,7 +43,7 @@ set(CONF_SECURITY_MONITOR_EXE_PATH "${WEBOS_INSTALL_BINDIR}/ls-monitor")
 set(CONF_SECURITY_MONITOR_PUB_EXE_PATH "${WEBOS_INSTALL_BINDIR}/ls-monitor-pub")
 set(CONF_SECURITY_SYSMGR_EXE_PATH "${WEBOS_INSTALL_BINDIR}/LunaSysMgr")
 set(CONF_SECURITY_WEBAPPMGR_EXE_PATH "${WEBOS_INSTALL_BINDIR}/WebAppMgr")
-set(CONF_SECURITY_WEBAPPMGR2_EXE_PATH "${WEBOS_INSTALL_BINDIR}/QtWebProcess")
+set(CONF_SECURITY_WEBAPPMGR2_EXE_PATH "${WEBOS_INSTALL_LIBDIR}/qt5/libexec/QtWebProcess")
 set(CONF_SECURITY_JS_SERVICE_EXE_PATH "js")
 set(CONF_SECURITY_MOJO_APP_EXE_PATH "mojo-app")
 set(CONF_SECURITY_MOJO_APPS_ALLOW_OUTBOUND_BY_DEFAULT "true")
-- 
1.8.1.2

