From ac02746ce18cd01ece4afc597b04d5a08af731ff Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 17 Dec 2022 17:34:37 +0000
Subject: [PATCH] Avoid crash if no interface found

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 src/technology.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/technology.c b/src/technology.c
index ff54059..600d8db 100644
--- a/src/technology.c
+++ b/src/technology.c
@@ -604,6 +604,10 @@ static void append_interfaces(DBusMessageIter *iter, void *user_data)
 	for (list = technology->device_list; list; list = list->next) {
 		struct connman_device *device = list->data;
 		const char *iface = connman_device_get_interface(device);
+		if (!iface) {
+			connman_error("Failed to get interface !!");
+			continue;
+		}
 
 		dbus_message_iter_append_basic(iter, DBUS_TYPE_STRING, &iface);
 	}
