From 5c555763925234af9056db80519a4c5f7283dd71 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Thu, 28 Dec 2023 10:48:50 +0100
Subject: [PATCH] com.webos.service.location: Add value for LuneOS

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS specific]

 files/etc/wsp.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/files/etc/wsp.conf b/files/etc/wsp.conf
index 96d245d..03e95b5 100644
--- a/files/etc/wsp.conf
+++ b/files/etc/wsp.conf
@@ -1,6 +1,6 @@
 {
         "wpsName": "google",
-        "apiKey": "",
+        "apiKey": "QUl6YVN5Qk9yaGJmbXYtMV9nMkJLMkFLXzNSNXEwcmlPLWU1Y19BCg==",
         "services": [
                 {
                         "features": ["geocode", "reverseGeocode"],
