From 4d3b675d1003a701855d883be24b8265891eb026 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 13 May 2019 18:14:24 +0000
Subject: [PATCH] ls-hubd: print more useful error for outbound permission
 error

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/ls-hubd/security.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ls-hubd/security.cpp b/src/ls-hubd/security.cpp
index 7046ac9..8af9de6 100644
--- a/src/ls-hubd/security.cpp
+++ b/src/ls-hubd/security.cpp
@@ -696,6 +696,11 @@ LSHubIsClientAllowedOutbound(_LSTransportClient *sender_client,
     {
         return true;
     }
+    else
+    {
+        _LSHubPrintPermissionsMessage(sender_client, sender_service_name, dest_service_name,
+                                      false, g_conf_security_enabled);
+    }
 
     return !g_conf_security_enabled;
 }
