From 8db553f7943b758f67d954fe02dcb9e526024451 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 2 Mar 2024 10:17:49 +0000
Subject: [PATCH] _LSCheckProvidedTrustedGroups: reduce log noise

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>

Upstream-Status: Pending
---
 src/libluna-service2/base.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libluna-service2/base.c b/src/libluna-service2/base.c
index 6fadf0d..245d8e7 100644
--- a/src/libluna-service2/base.c
+++ b/src/libluna-service2/base.c
@@ -659,7 +659,7 @@ LSMessageHandlerResult _LSCheckProvidedTrustedGroups(LSHandle *sh,
         jvalue_ref providedGroupTrustLevel = NULL;
         jvalue_ref providedGroupsRef = NULL;
 
-        LOG_LS_INFO(MSGID_LS_NOT_AN_ERROR, 0,"Enhanced ACG \n");
+        LOG_LS_DEBUG(MSGID_LS_NOT_AN_ERROR, 0,"Enhanced ACG \n");
         // prepare full methods name for pattern matching
         //char *full_name = g_build_path("/", category_path, m->name, NULL);
         const LSTransportTrustLevelGroupBitmask *TrustLevel_bitmask = NULL;
@@ -688,7 +688,7 @@ LSMessageHandlerResult _LSCheckProvidedTrustedGroups(LSHandle *sh,
                     {
                         if(TrustLevel_bitmask->trustLevel_group_bitmask)
                         {
-                            LOG_LS_INFO(MSGID_LS_NOT_AN_ERROR, 0, "[%s] found group bit mask : %d \n", __func__,
+                            LOG_LS_DEBUG(MSGID_LS_NOT_AN_ERROR, 0, "[%s] found group bit mask : %d \n", __func__,
                                                *TrustLevel_bitmask->trustLevel_group_bitmask);
                             providedGroupTrustLevel = LSTransportGetTrustFromMask(sh->transport,
                                                           TrustLevel_bitmask->trustLevel_group_bitmask);
