From f22ef12864534fd2d6369b4c5b20f77bb7726396 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sat, 12 Jul 2014 15:44:15 +0200
Subject: [PATCH] gril: don't exit when not being able to initialize modem

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 gril/gril.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gril/gril.c b/gril/gril.c
index bd44f8b..7de81fa 100644
--- a/gril/gril.c
+++ b/gril/gril.c
@@ -895,8 +895,7 @@ static struct ril_s *create_ril(const char *sock_path)
 	return ril;
 
 error:
-	ofono_error("Exiting...");
-	exit(EXIT_FAILURE);
+	return NULL;
 }
 
 static struct ril_notify *ril_notify_create(struct ril_s *ril,
-- 
1.9.1

