From 04e5f935a3b71796df9590c3666c73d92e20139f Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sat, 17 Aug 2013 12:09:38 +0200
Subject: [PATCH] Disable support for android audiosystem

Upstream-Status: Inappropiate [disable feature]

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 Makefile.am                  |  2 +-
 drivers/rilmodem/voicecall.c | 11 ++++++++---
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 404fcc8..908721f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -537,7 +537,7 @@ src_ofonod_SOURCES = $(gdbus_sources) $(builtin_sources) src/ofono.ver \
 			src/cdma-sms.c src/private-network.c src/cdma-netreg.c \
 			src/cdma-provision.c src/handsfree.c
 
-src_ofonod_LDADD = $(builtin_libadd) @GLIB_LIBS@ @DBUS_LIBS@ -ldl -lwaudio
+src_ofonod_LDADD = $(builtin_libadd) @GLIB_LIBS@ @DBUS_LIBS@ -ldl
 
 src_ofonod_LDFLAGS = -Wl,--export-dynamic \
 				-Wl,--version-script=$(srcdir)/src/ofono.ver
diff --git a/drivers/rilmodem/voicecall.c b/drivers/rilmodem/voicecall.c
index 473824f..50a4a6a 100644
--- a/drivers/rilmodem/voicecall.c
+++ b/drivers/rilmodem/voicecall.c
@@ -32,9 +32,6 @@
 
 #include <glib.h>
 
-/* For AudioFlinger settings */
-#include <waudio.h>
-
 #include <ofono/log.h>
 #include <ofono/modem.h>
 #include <ofono/voicecall.h>
@@ -79,6 +76,7 @@ struct change_state_req {
 	int affected_types;
 };
 
+#if 0
 static void audioflinger_set_call_mode()
 {
 	char parameter[20];
@@ -120,6 +118,7 @@ static void audioflinger_set_normal_mode()
 			break;
 	}
 }
+#endif
 
 static void clcc_poll_cb(struct ril_msg *message, gpointer user_data)
 {
@@ -203,10 +202,12 @@ static void clcc_poll_cb(struct ril_msg *message, gpointer user_data)
 		}
 	}
 
+#if 0
 	/* No other calls, get audioflinger into normal state */
 	if (vd->calls && !calls) {
 		audioflinger_set_normal_mode();
 	}
+#endif
 
 	g_slist_foreach(vd->calls, (GFunc) g_free, NULL);
 	g_slist_free(vd->calls);
@@ -336,7 +337,9 @@ static void rild_cb(struct ril_msg *message, gpointer user_data)
 		vd->clcc_source = g_timeout_add(POLL_CLCC_INTERVAL,
 						poll_clcc, vc);
 
+#if 0
 	audioflinger_set_call_mode();
+#endif
 
 out:
 	cb(&error, cbd->data);
@@ -450,7 +453,9 @@ static void ril_answer(struct ofono_voicecall *vc,
 
 	g_ril_print_request_no_args(vd->ril, ret, request);
 
+#if 0
 	audioflinger_set_call_mode();
+#endif
 }
 
 static void ril_send_dtmf(struct ofono_voicecall *vc, const char *dtmf,
-- 
1.8.1.2

