From 2336a2c542f1feaa9aff045fe7522631f108af69 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Thu, 20 Nov 2014 19:21:13 +0100
Subject: [PATCH] Enable the various modem plugins we support again

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 ofono/Makefile.am | 2 --
 1 file changed, 2 deletions(-)

diff --git a/ofono/Makefile.am b/ofono/Makefile.am
index 30aa90d..f387f9d 100644
--- a/ofono/Makefile.am
+++ b/ofono/Makefile.am
@@ -386,7 +386,6 @@ builtin_sources += drivers/cdmamodem/cdmamodem.h \
 			drivers/cdmamodem/connman.c
 endif
 
-if !RILMODEM
 builtin_modules += g1
 builtin_sources += plugins/g1.c
 
@@ -458,7 +457,6 @@ builtin_sources += plugins/samsung.c
 
 builtin_modules += sim900
 builtin_sources += plugins/sim900.c
-endif
 
 builtin_modules += connman
 builtin_sources += plugins/connman.c
-- 
1.9.1

