From 6e511886ff096f5364772a1fe9c806f99415a0af Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 30 Apr 2016 18:53:20 +0200
Subject: [PATCH] fd_fdinfo_mnt_id: disable fdinfo stat

Upstream-Status: Pending
---
 src/basic/mountpoint-util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/basic/mountpoint-util.c b/src/basic/mountpoint-util.c
index 8c836a1b74..98f5e07dc9 100644
--- a/src/basic/mountpoint-util.c
+++ b/src/basic/mountpoint-util.c
@@ -114,6 +114,7 @@ static int fd_fdinfo_mnt_id(int fd, const char *filename, int flags, int *ret_mn
         }
 
         r = read_full_virtual_file(path, &fdinfo, NULL);
+        r = -ENOENT;
         if (r == -ENOENT) /* The fdinfo directory is a relatively new addition */
                 return -EOPNOTSUPP;
         if (r < 0)
