From feec8acb947003fcb64d85bcef1b8f5b23c261ab Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 30 Apr 2016 18:53:20 +0200
Subject: [PATCH] fd_fdinfo_mnt_id: disable fdinfo stat

Upstream-Status: Pending
---
 src/basic/mountpoint-util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/basic/mountpoint-util.c b/src/basic/mountpoint-util.c
index 5af0896ad6..395574f357 100644
--- a/src/basic/mountpoint-util.c
+++ b/src/basic/mountpoint-util.c
@@ -122,6 +122,7 @@ static int fd_fdinfo_mnt_id(int fd, const char *filename, int flags, int *ret_mn
         }
 
         r = read_full_virtual_file(path, &fdinfo, NULL);
+        r = -ENOENT;
         if (r == -ENOENT) /* The fdinfo directory is a relatively new addition */
                 return proc_mounted() > 0 ? -EOPNOTSUPP : -ENOSYS;
         if (r < 0)
