From 59c8e00a68847a244a1c1f9b227bcb8fde087de8 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 30 Apr 2016 18:53:20 +0200
Subject: [PATCH] fd_fdinfo_mnt_id: disable fdinfo stat

---
 src/basic/mount-util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/basic/mount-util.c b/src/basic/mount-util.c
index ebe41a4c6..b85957c19 100644
--- a/src/basic/mount-util.c
+++ b/src/basic/mount-util.c
@@ -117,6 +117,7 @@ static int fd_fdinfo_mnt_id(int fd, const char *filename, int flags, int *mnt_id
         }
 
         r = read_full_file(path, &fdinfo, NULL);
+        r = -ENOENT;
         if (r == -ENOENT) /* The fdinfo directory is a relatively new addition */
                 return -EOPNOTSUPP;
         if (r < 0)
