From adcc1b8c4822c145be27720dc91d314a72300524 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 29 Apr 2017 14:41:21 +0000
Subject: [PATCH] Disable ProtectHome and ProtectSystem for old kernels

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 src/core/load-fragment.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/core/load-fragment.c b/src/core/load-fragment.c
index cbc8268..faeaecd 100644
--- a/src/core/load-fragment.c
+++ b/src/core/load-fragment.c
@@ -3861,7 +3861,8 @@ int config_parse_protect_home(
         /* Our enum shall be a superset of booleans, hence first try
          * to parse as boolean, and then as enum */
 
-        k = parse_boolean(rvalue);
+        /* k = parse_boolean(rvalue); */
+        k = 0; /* Disable protection for old kernels */
         if (k > 0)
                 c->protect_home = PROTECT_HOME_YES;
         else if (k == 0)
@@ -3904,7 +3905,8 @@ int config_parse_protect_system(
         /* Our enum shall be a superset of booleans, hence first try
          * to parse as boolean, and then as enum */
 
-        k = parse_boolean(rvalue);
+        /* k = parse_boolean(rvalue); */
+        k = 0; /* Disable protection for old kernels */
         if (k > 0)
                 c->protect_system = PROTECT_SYSTEM_YES;
         else if (k == 0)
-- 
2.7.4

