From a598480a7ebf8a5262553274bb816c3c4728e152 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Wed, 4 Jun 2025 11:51:57 +0200
Subject: [PATCH] Add relaunch support for legacy enyo apps

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
Upstream-Status: Inappropriate [LuneOS specific]
---
 src/core/web_page_base.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/core/web_page_base.cc b/src/core/web_page_base.cc
index fbba2c2..c7f94ef 100644
--- a/src/core/web_page_base.cc
+++ b/src/core/web_page_base.cc
@@ -256,6 +256,12 @@ void WebPageBase::SendRelaunchEvent() {
       << "    console.log('[WAM] fires webOSRelaunch event');"
       << "    var launchEvent=new CustomEvent('webOSRelaunch', { detail: "
       << detail << " });" << "    document.dispatchEvent(launchEvent);"
+
+      << "    console.log('[WAM] call Mojo.relaunch() for legacy apps');"
+      << "    if(typeof Mojo !== 'undefined') {"
+      << "      PalmSystem.launchParams = JSON.stringify(" << detail << ");"
+      << "      Mojo.relaunch();"
+      << "    }"
       << "}, 1);";
   EvaluateJavaScript(relaunch_event.str().c_str());
 }
