From f4cb07028c65e360044f2007772dca2e6d36fcbd Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 3 Apr 2022 10:11:39 +0000
Subject: [PATCH] Add relaunch support for legacy enyo apps

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/core/web_page_base.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/core/web_page_base.cc b/src/core/web_page_base.cc
index e8d1fb2..6028e75 100644
--- a/src/core/web_page_base.cc
+++ b/src/core/web_page_base.cc
@@ -274,6 +274,12 @@ void WebPageBase::SendRelaunchEvent() {
       << "    var launchEvent=new CustomEvent('webOSRelaunch', { detail: "
       << detail << " });"
       << "    document.dispatchEvent(launchEvent);"
+
+      << "    console.log('[WAM] call Mojo.relaunch() for legacy apps');"
+      << "    if(typeof Mojo !== 'undefined') {"
+      << "      PalmSystem.launchParams = JSON.stringify(" << detail << ");"
+      << "      Mojo.relaunch();"
+      << "    }"
       << "}, 1);";
   EvaluateJavaScript(relaunch_event.str().c_str());
 }
