From 904d623b432b48623101e488e32c4ae627b23ff9 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Fri, 9 May 2014 09:36:31 +0200
Subject: [PATCH] Add com.palm.mediapermissions service as admin for maindb

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 files/conf/maindb.conf | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/files/conf/maindb.conf b/files/conf/maindb.conf
index 7a8403c..8ed618c 100644
--- a/files/conf/maindb.conf
+++ b/files/conf/maindb.conf
@@ -16,7 +16,8 @@
 			{"type":"db.role","object":"admin","caller":"com.palm.service.backup","operations":{"*":"allow"}},
 			{"type":"db.role","object":"admin","caller":"com.palm.odd.service","operations":{"*":"allow"}},
 			{"type":"db.role","object":"admin","caller":"com.palm.service.migrationscript","operations":{"*":"allow"}},
-			{"type":"db.role","object":"admin","caller":"com.palm.spacecadet","operations":{"*":"allow"}}
+			{"type":"db.role","object":"admin","caller":"com.palm.spacecadet","operations":{"*":"allow"}},
+			{"type":"db.role","object":"admin","caller":"com.palm.mediapermissions","operations":{"*":"allow"}}
 		],
 		"quotas" : [
                         {"owner":"*","size":10485760},
-- 
1.9.1

