diff --git a/CMakeLists.txt b/CMakeLists.txt
index 89f494f..74ccff3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ endif()
 MESSAGE(STATUS "BDB LIB ${BDB}")
 
 webos_build_pkgconfig(files/pkgconfig/mojodb)
-webos_add_compiler_flags(ALL -g -O2 -Wall -fPIC -Wconversion -Werror
+webos_add_compiler_flags(ALL -g -O2 -Wall -fPIC -Wconversion
                          -Wno-unused-but-set-variable -Wno-unused-variable
                          -fno-exceptions -fno-rtti -DNDEBUG -fexpensive-optimizations
                          -frename-registers -DMOJ_LINUX -fno-strict-aliasing
